<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=author content="Genomics Aotearoa & NeSI"><link href=https://genomicsaotearoa.github.io/RNA-seq-workshop/4_mappingANDcount/ rel=canonical><link rel=icon href=../Images/nesi_ga.png><meta name=generator content="mkdocs-1.4.1, mkdocs-material-8.3.9"><title>Mapping and Counting - RNA-Seq Data Analysis</title><link rel=stylesheet href=../assets/stylesheets/main.1d29e8d0.min.css><link rel=stylesheet href=../assets/stylesheets/palette.cbb835fc.min.css><style>:root{--md-admonition-icon--note:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M2.5 7.775V2.75a.25.25 0 0 1 .25-.25h5.025a.25.25 0 0 1 .177.073l6.25 6.25a.25.25 0 0 1 0 .354l-5.025 5.025a.25.25 0 0 1-.354 0l-6.25-6.25a.25.25 0 0 1-.073-.177zm-1.5 0V2.75C1 1.784 1.784 1 2.75 1h5.025c.464 0 .91.184 1.238.513l6.25 6.25a1.75 1.75 0 0 1 0 2.474l-5.026 5.026a1.75 1.75 0 0 1-2.474 0l-6.25-6.25A1.75 1.75 0 0 1 1 7.775zM6 5a1 1 0 1 0 0 2 1 1 0 0 0 0-2z"/></svg>');--md-admonition-icon--abstract:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M2.5 1.75a.25.25 0 0 1 .25-.25h8.5a.25.25 0 0 1 .25.25v7.736a.75.75 0 1 0 1.5 0V1.75A1.75 1.75 0 0 0 11.25 0h-8.5A1.75 1.75 0 0 0 1 1.75v11.5c0 .966.784 1.75 1.75 1.75h3.17a.75.75 0 0 0 0-1.5H2.75a.25.25 0 0 1-.25-.25V1.75zM4.75 4a.75.75 0 0 0 0 1.5h4.5a.75.75 0 0 0 0-1.5h-4.5zM4 7.75A.75.75 0 0 1 4.75 7h2a.75.75 0 0 1 0 1.5h-2A.75.75 0 0 1 4 7.75zm11.774 3.537a.75.75 0 0 0-1.048-1.074L10.7 14.145 9.281 12.72a.75.75 0 0 0-1.062 1.058l1.943 1.95a.75.75 0 0 0 1.055.008l4.557-4.45z"/></svg>');--md-admonition-icon--info:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8 1.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13zM0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8zm6.5-.25A.75.75 0 0 1 7.25 7h1a.75.75 0 0 1 .75.75v2.75h.25a.75.75 0 0 1 0 1.5h-2a.75.75 0 0 1 0-1.5h.25v-2h-.25a.75.75 0 0 1-.75-.75zM8 6a1 1 0 1 0 0-2 1 1 0 0 0 0 2z"/></svg>');--md-admonition-icon--tip:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M3.499.75a.75.75 0 0 1 1.5 0v.996C5.9 2.903 6.793 3.65 7.662 4.376l.24.202c-.036-.694.055-1.422.426-2.163C9.1.873 10.794-.045 12.622.26 14.408.558 16 1.94 16 4.25c0 1.278-.954 2.575-2.44 2.734l.146.508.065.22c.203.701.412 1.455.476 2.226.142 1.707-.4 3.03-1.487 3.898C11.714 14.671 10.27 15 8.75 15h-6a.75.75 0 0 1 0-1.5h1.376a4.489 4.489 0 0 1-.563-1.191 3.833 3.833 0 0 1-.05-2.063 4.636 4.636 0 0 1-2.025-.293.75.75 0 1 1 .525-1.406c1.357.507 2.376-.006 2.698-.318l.009-.01a.748.748 0 0 1 1.06 0 .75.75 0 0 1-.012 1.074c-.912.92-.992 1.835-.768 2.586.221.74.745 1.337 1.196 1.621H8.75c1.343 0 2.398-.296 3.074-.836.635-.507 1.036-1.31.928-2.602-.05-.603-.216-1.224-.422-1.93l-.064-.221c-.12-.407-.246-.84-.353-1.29a2.404 2.404 0 0 1-.507-.441 3.063 3.063 0 0 1-.633-1.248.75.75 0 0 1 1.455-.364c.046.185.144.436.31.627.146.168.353.305.712.305.738 0 1.25-.615 1.25-1.25 0-1.47-.95-2.315-2.123-2.51-1.172-.196-2.227.387-2.706 1.345-.46.92-.27 1.774.019 3.062l.042.19a.753.753 0 0 1 .01.05c.348.443.666.949.94 1.553a.75.75 0 1 1-1.365.62c-.553-1.217-1.32-1.94-2.3-2.768a85.08 85.08 0 0 0-.317-.265c-.814-.68-1.75-1.462-2.692-2.619a3.74 3.74 0 0 0-1.023.88c-.406.495-.663 1.036-.722 1.508.116.122.306.21.591.239.388.038.797-.06 1.032-.19a.75.75 0 0 1 .728 1.31c-.515.287-1.23.439-1.906.373-.682-.067-1.473-.38-1.879-1.193L.75 5.677V5.5c0-.984.48-1.94 1.077-2.664.46-.559 1.05-1.055 1.673-1.353V.75z"/></svg>');--md-admonition-icon--success:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M13.78 4.22a.75.75 0 0 1 0 1.06l-7.25 7.25a.75.75 0 0 1-1.06 0L2.22 9.28a.75.75 0 0 1 1.06-1.06L6 10.94l6.72-6.72a.75.75 0 0 1 1.06 0z"/></svg>');--md-admonition-icon--question:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8 1.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13zM0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8zm9 3a1 1 0 1 1-2 0 1 1 0 0 1 2 0zM6.92 6.085c.081-.16.19-.299.34-.398.145-.097.371-.187.74-.187.28 0 .553.087.738.225A.613.613 0 0 1 9 6.25c0 .177-.04.264-.077.318a.956.956 0 0 1-.277.245c-.076.051-.158.1-.258.161l-.007.004a7.728 7.728 0 0 0-.313.195 2.416 2.416 0 0 0-.692.661.75.75 0 0 0 1.248.832.956.956 0 0 1 .276-.245 6.3 6.3 0 0 1 .26-.16l.006-.004c.093-.057.204-.123.313-.195.222-.149.487-.355.692-.662.214-.32.329-.702.329-1.15 0-.76-.36-1.348-.863-1.725A2.76 2.76 0 0 0 8 4c-.631 0-1.155.16-1.572.438-.413.276-.68.638-.849.977a.75.75 0 1 0 1.342.67z"/></svg>');--md-admonition-icon--warning:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.22 1.754a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368L8.22 1.754zm-1.763-.707c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575L6.457 1.047zM9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0zm-.25-5.25a.75.75 0 0 0-1.5 0v2.5a.75.75 0 0 0 1.5 0v-2.5z"/></svg>');--md-admonition-icon--failure:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M3.404 12.596a6.5 6.5 0 1 1 9.192-9.192 6.5 6.5 0 0 1-9.192 9.192zM2.344 2.343a8 8 0 1 0 11.313 11.314A8 8 0 0 0 2.343 2.343zM6.03 4.97a.75.75 0 0 0-1.06 1.06L6.94 8 4.97 9.97a.75.75 0 1 0 1.06 1.06L8 9.06l1.97 1.97a.75.75 0 1 0 1.06-1.06L9.06 8l1.97-1.97a.75.75 0 1 0-1.06-1.06L8 6.94 6.03 4.97z"/></svg>');--md-admonition-icon--danger:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M10.561 1.5a.016.016 0 0 0-.01.004L3.286 8.571A.25.25 0 0 0 3.462 9H6.75a.75.75 0 0 1 .694 1.034l-1.713 4.188 6.982-6.793A.25.25 0 0 0 12.538 7H9.25a.75.75 0 0 1-.683-1.06l2.008-4.418.003-.006a.02.02 0 0 0-.004-.009.02.02 0 0 0-.006-.006L10.56 1.5zM9.504.43a1.516 1.516 0 0 1 2.437 1.713L10.415 5.5h2.123c1.57 0 2.346 1.909 1.22 3.004l-7.34 7.142a1.25 1.25 0 0 1-.871.354h-.302a1.25 1.25 0 0 1-1.157-1.723L5.633 10.5H3.462c-1.57 0-2.346-1.909-1.22-3.004L9.503.429z"/></svg>');--md-admonition-icon--bug:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M4.72.22a.75.75 0 0 1 1.06 0l1 .999a3.492 3.492 0 0 1 2.441 0l.999-1a.75.75 0 1 1 1.06 1.061l-.775.776c.616.63.995 1.493.995 2.444v.327c0 .1-.009.197-.025.292.408.14.764.392 1.029.722l1.968-.787a.75.75 0 0 1 .556 1.392L13 7.258V9h2.25a.75.75 0 0 1 0 1.5H13v.5c0 .409-.049.806-.141 1.186l2.17.868a.75.75 0 0 1-.557 1.392l-2.184-.873A4.997 4.997 0 0 1 8 16a4.997 4.997 0 0 1-4.288-2.427l-2.183.873a.75.75 0 0 1-.558-1.392l2.17-.868A5.013 5.013 0 0 1 3 11v-.5H.75a.75.75 0 0 1 0-1.5H3V7.258L.971 6.446a.75.75 0 0 1 .558-1.392l1.967.787c.265-.33.62-.583 1.03-.722a1.684 1.684 0 0 1-.026-.292V4.5c0-.951.38-1.814.995-2.444L4.72 1.28a.75.75 0 0 1 0-1.06zM6.173 5h3.654A.173.173 0 0 0 10 4.827V4.5a2 2 0 1 0-4 0v.327c0 .096.077.173.173.173zM5.25 6.5a.75.75 0 0 0-.75.75V11a3.5 3.5 0 1 0 7 0V7.25a.75.75 0 0 0-.75-.75h-5.5z"/></svg>');--md-admonition-icon--example:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M5 5.782V2.5h-.25a.75.75 0 0 1 0-1.5h6.5a.75.75 0 0 1 0 1.5H11v3.282l3.666 5.76C15.619 13.04 14.543 15 12.767 15H3.233c-1.776 0-2.852-1.96-1.899-3.458L5 5.782zM9.5 2.5h-3V6a.75.75 0 0 1-.117.403L4.73 9h6.54L9.617 6.403A.75.75 0 0 1 9.5 6V2.5zm-6.9 9.847L3.775 10.5h8.45l1.175 1.847a.75.75 0 0 1-.633 1.153H3.233a.75.75 0 0 1-.633-1.153z"/></svg>');--md-admonition-icon--quote:url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M1.75 2.5a.75.75 0 0 0 0 1.5h10.5a.75.75 0 0 0 0-1.5H1.75zm4 5a.75.75 0 0 0 0 1.5h8.5a.75.75 0 0 0 0-1.5h-8.5zm0 5a.75.75 0 0 0 0 1.5h8.5a.75.75 0 0 0 0-1.5h-8.5zM2.5 7.75a.75.75 0 0 0-1.5 0v6a.75.75 0 0 0 1.5 0v-6z"/></svg>');}</style><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style><script>__md_scope=new URL("..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script></head> <body dir=ltr data-md-color-scheme data-md-color-primary=green data-md-color-accent> <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#mapping-and-counting class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=.. title="RNA-Seq Data Analysis" class="md-header__button md-logo" aria-label="RNA-Seq Data Analysis" data-md-component=logo> <img src=../Images/nesi_ga.png alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> RNA-Seq Data Analysis </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> Mapping and Counting </span> </div> </div> </div> <form class=md-header__option data-md-component=palette> <input class=md-option data-md-color-media=(prefers-color-scheme) data-md-color-scheme data-md-color-primary=green data-md-color-accent aria-label="Switch to light mode" type=radio name=__palette id=__palette_1> <label class="md-header__button md-icon" title="Switch to light mode" for=__palette_3 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"/></svg> </label> <input class=md-option data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme=default data-md-color-primary=green data-md-color-accent aria-label="Switch to dark mode" type=radio name=__palette id=__palette_2> <label class="md-header__button md-icon" title="Switch to dark mode" for=__palette_1 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg> </label> <input class=md-option data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme=slate data-md-color-primary=green data-md-color-accent aria-label="Switch to system preference" type=radio name=__palette id=__palette_3> <label class="md-header__button md-icon" title="Switch to system preference" for=__palette_2 hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg> </label> </form> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <button type=reset class="md-search__icon md-icon" aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/GenomicsAotearoa/RNA-seq-workshop.git title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> GenomicsAotearoa/RNA-seq-workshop </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=.. title="RNA-Seq Data Analysis" class="md-nav__button md-logo" aria-label="RNA-Seq Data Analysis" data-md-component=logo> <img src=../Images/nesi_ga.png alt=logo> </a> RNA-Seq Data Analysis </label> <div class=md-nav__source> <a href=https://github.com/GenomicsAotearoa/RNA-seq-workshop.git title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> GenomicsAotearoa/RNA-seq-workshop </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=.. class=md-nav__link> Home </a> </li> <li class=md-nav__item> <a href=../1_background/ class=md-nav__link> RNA-seq Workflow </a> </li> <li class=md-nav__item> <a href=../2_qualityassessment/ class=md-nav__link> Quality control of the sequencing data. </a> </li> <li class=md-nav__item> <a href=../3_trimmingfiltering/ class=md-nav__link> Cleaning Reads </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" data-md-toggle=toc type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> Mapping and Counting <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> Mapping and Counting </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#preparation-of-the-genome class=md-nav__link> Preparation of the genome </a> </li> <li class=md-nav__item> <a href=#alignment-on-the-genome class=md-nav__link> Alignment on the genome </a> </li> <li class=md-nav__item> <a href=#converting-sam-files-to-bam-files class=md-nav__link> Converting SAM files to BAM files </a> <nav class=md-nav aria-label="Converting SAM files to BAM files"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#a-quick-look-into-the-sam-file class=md-nav__link> A quick look into the sam file </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#some-stats-on-your-mapping class=md-nav__link> Some stats on your mapping: </a> </li> <li class=md-nav__item> <a href=#multiqc-hisat2-output class=md-nav__link> MultiQC: HISAT2 output </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../5_rnaseq-diffexp/ class=md-nav__link> Differential Expression Analysis </a> </li> <li class=md-nav__item> <a href=../6_over-representation_analysis/ class=md-nav__link> Over-representation analysis (Gene Ontology) </a> </li> <li class=md-nav__item> <a href=../supp_1/ class=md-nav__link> Supplementary 1 - A Guide to RNA-Seq </a> </li> <li class=md-nav__item> <a href=../supp_2/ class=md-nav__link> Supplementary 2 -Lecture notes : Introduction To Differential Expression Analysis </a> </li> <li class=md-nav__item> <a href=../supp_3/ class=md-nav__link> Supplementary 3 - Lecture notes : Annotation Data & Gene Set Analysis </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#preparation-of-the-genome class=md-nav__link> Preparation of the genome </a> </li> <li class=md-nav__item> <a href=#alignment-on-the-genome class=md-nav__link> Alignment on the genome </a> </li> <li class=md-nav__item> <a href=#converting-sam-files-to-bam-files class=md-nav__link> Converting SAM files to BAM files </a> <nav class=md-nav aria-label="Converting SAM files to BAM files"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#a-quick-look-into-the-sam-file class=md-nav__link> A quick look into the sam file </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#some-stats-on-your-mapping class=md-nav__link> Some stats on your mapping: </a> </li> <li class=md-nav__item> <a href=#multiqc-hisat2-output class=md-nav__link> MultiQC: HISAT2 output </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/GenomicsAotearoa/RNA-seq-workshop.git/edit/master/docs/4_mappingANDcount.md title="Edit this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg> </a> <h1 id=mapping-and-counting>Mapping and Counting<a class=headerlink href=#mapping-and-counting title="Permanent link">&para;</a></h1> <div class="admonition info"> <p class=admonition-title>Objectives</p> <ul> <li>Learn how to index a reference genome</li> <li>Align sequence reads to an indexed reference genome</li> <li>Generate read counts for each exon in the genome</li> </ul> </div> <p><center> <img alt=imge src=../theme_images/mapping.png width=300> </center></p> <p>RNA-seq generate gene expression information by quantifying the number of transcripts (per gene) in a sample. This is accomplished by counting the number of transcripts that have been sequenced - the more active a gene is, the more transcripts will be in a sample, and the more reads will be generated from that transcript.</p> <p>For RNA-seq, we need to align or map each read back to the genome, to see which gene produced it. - Highly expressed genes will generate lots of transcripts, so there will be lots of reads that map back to the position of that transcript in the genome. - The per-gene data we work with in an RNA-seq experiment are counts: the number of reads from each sample that originated from that gene.</p> <h2 id=preparation-of-the-genome>Preparation of the genome<a class=headerlink href=#preparation-of-the-genome title="Permanent link">&para;</a></h2> <p>To be able to map (align) sequencing reads on the genome, the genome needs to be indexed first. In this workshop we will use <a href=https://www.nature.com/articles/nmeth.3317>HISAT2</a>. Note for speed reason, the reads will be aligned on the chr5 of the Yeast genome.</p> <div class=highlight><pre><span></span><code><a id=__codelineno-0-1 name=__codelineno-0-1 href=#__codelineno-0-1></a>$ <span class=nb>cd</span> /home/<span class=nv>$USER</span>/RNA_seq/Genome
<a id=__codelineno-0-2 name=__codelineno-0-2 href=#__codelineno-0-2></a>
<a id=__codelineno-0-3 name=__codelineno-0-3 href=#__codelineno-0-3></a><span class=c1>#to list what is in your directory:</span>
<a id=__codelineno-0-4 name=__codelineno-0-4 href=#__codelineno-0-4></a>$ ls /home/<span class=nv>$USER</span>/RNA_seq/Genome
<a id=__codelineno-0-5 name=__codelineno-0-5 href=#__codelineno-0-5></a>Saccharomyces_cerevisiae.R64-1-1.99.gtf  Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa
<a id=__codelineno-0-6 name=__codelineno-0-6 href=#__codelineno-0-6></a>
<a id=__codelineno-0-7 name=__codelineno-0-7 href=#__codelineno-0-7></a>$ module load HISAT2/2.2.0-gimkl-2020a
<a id=__codelineno-0-8 name=__codelineno-0-8 href=#__codelineno-0-8></a>
<a id=__codelineno-0-9 name=__codelineno-0-9 href=#__codelineno-0-9></a><span class=c1># index file:</span>
<a id=__codelineno-0-10 name=__codelineno-0-10 href=#__codelineno-0-10></a>$ hisat2-build -p <span class=m>4</span> -f Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa Saccharomyces_cerevisiae.R64-1-1.dna.toplevel
<a id=__codelineno-0-11 name=__codelineno-0-11 href=#__codelineno-0-11></a>
<a id=__codelineno-0-12 name=__codelineno-0-12 href=#__codelineno-0-12></a><span class=c1>#list what is in the directory:</span>
<a id=__codelineno-0-13 name=__codelineno-0-13 href=#__codelineno-0-13></a>$ ls /home/<span class=nv>$USER</span>/RNA_seq/Genome
<a id=__codelineno-0-14 name=__codelineno-0-14 href=#__codelineno-0-14></a>Saccharomyces_cerevisiae.R64-1-1.99.gtf              Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.4.ht2  Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.8.ht2
<a id=__codelineno-0-15 name=__codelineno-0-15 href=#__codelineno-0-15></a>Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.1.ht2  Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.5.ht2  Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa
<a id=__codelineno-0-16 name=__codelineno-0-16 href=#__codelineno-0-16></a>Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.2.ht2  Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.6.ht2
<a id=__codelineno-0-17 name=__codelineno-0-17 href=#__codelineno-0-17></a>Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.3.ht2  Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.7.ht2
</code></pre></div> <div class="admonition quote"> <ul> <li><strong>-p</strong> number of threads</li> <li><strong>-f</strong> fasta file</li> </ul> </div> <p>How many files were created during the indexing process?</p> <h2 id=alignment-on-the-genome>Alignment on the genome<a class=headerlink href=#alignment-on-the-genome title="Permanent link">&para;</a></h2> <p>Now that the genome is prepared. Sequencing reads can be aligned.</p> <p>Information required:</p> <ul> <li>Where the sequence information is stored (e.g. fastq files ...) ?</li> <li>What kind of sequencing: Single End or Paired end ?</li> <li>Where are stored the indexes and the genome? </li> <li> <p>Where will the mapping files be stored?</p> </li> <li> <p>Now, lets move one folder up (into the RNA_seq folder):</p> </li> </ul> <div class=highlight><pre><span></span><code><a id=__codelineno-1-1 name=__codelineno-1-1 href=#__codelineno-1-1></a>$ <span class=nb>cd</span> ..
<a id=__codelineno-1-2 name=__codelineno-1-2 href=#__codelineno-1-2></a>
<a id=__codelineno-1-3 name=__codelineno-1-3 href=#__codelineno-1-3></a>$ ls
<a id=__codelineno-1-4 name=__codelineno-1-4 href=#__codelineno-1-4></a>Genome  QC  Raw  Trimmed
</code></pre></div> <p>Let's map one of our sample to the genome</p> <div class=highlight><pre><span></span><code><a id=__codelineno-2-1 name=__codelineno-2-1 href=#__codelineno-2-1></a>$ hisat2 -x Genome/Saccharomyces_cerevisiae.R64-1-1.dna.toplevel -U Raw/SRR014335-chr1.fastq -S SRR014335.sam
<a id=__codelineno-2-2 name=__codelineno-2-2 href=#__codelineno-2-2></a><span class=m>125090</span> reads<span class=p>;</span> of these:
<a id=__codelineno-2-3 name=__codelineno-2-3 href=#__codelineno-2-3></a>  <span class=m>125090</span> <span class=o>(</span><span class=m>100</span>.00%<span class=o>)</span> were unpaired<span class=p>;</span> of these:
<a id=__codelineno-2-4 name=__codelineno-2-4 href=#__codelineno-2-4></a>    <span class=m>20537</span> <span class=o>(</span><span class=m>16</span>.42%<span class=o>)</span> aligned <span class=m>0</span> <span class=nb>times</span>
<a id=__codelineno-2-5 name=__codelineno-2-5 href=#__codelineno-2-5></a>    <span class=m>85066</span> <span class=o>(</span><span class=m>68</span>.00%<span class=o>)</span> aligned exactly <span class=m>1</span> <span class=nb>time</span>
<a id=__codelineno-2-6 name=__codelineno-2-6 href=#__codelineno-2-6></a>    <span class=m>19487</span> <span class=o>(</span><span class=m>15</span>.58%<span class=o>)</span> aligned &gt;1 <span class=nb>times</span>
<a id=__codelineno-2-7 name=__codelineno-2-7 href=#__codelineno-2-7></a><span class=m>83</span>.58% overall alignment rate
</code></pre></div> <div class="admonition quote"> <ul> <li><strong>-x</strong> The basename of the index for the reference genome. </li> <li><strong>-U</strong> Comma-separated list of files containing unpaired reads to be aligned</li> <li><strong>-S</strong> File to write SAM alignments to. By default, alignments are written to the “standard out” or “stdout” filehandle </li> </ul> </div> <p>Now we need to align all the rest of the samples.</p> <div class=highlight><pre><span></span><code><a id=__codelineno-3-1 name=__codelineno-3-1 href=#__codelineno-3-1></a>$ <span class=nb>pwd</span>
<a id=__codelineno-3-2 name=__codelineno-3-2 href=#__codelineno-3-2></a>/home/<span class=nv>$USER</span>/RNA_seq/
<a id=__codelineno-3-3 name=__codelineno-3-3 href=#__codelineno-3-3></a>
<a id=__codelineno-3-4 name=__codelineno-3-4 href=#__codelineno-3-4></a>$ mkdir Mapping
<a id=__codelineno-3-5 name=__codelineno-3-5 href=#__codelineno-3-5></a>
<a id=__codelineno-3-6 name=__codelineno-3-6 href=#__codelineno-3-6></a>$ ls
<a id=__codelineno-3-7 name=__codelineno-3-7 href=#__codelineno-3-7></a>Genome  Mapping  QC  Raw  SRR014335.sam  Trimmed
</code></pre></div> <p>let's use a for loop to process our samples:</p> <p><div class=highlight><pre><span></span><code><a id=__codelineno-4-1 name=__codelineno-4-1 href=#__codelineno-4-1></a>$ <span class=nb>cd</span> Trimmed
<a id=__codelineno-4-2 name=__codelineno-4-2 href=#__codelineno-4-2></a>
<a id=__codelineno-4-3 name=__codelineno-4-3 href=#__codelineno-4-3></a>$ ls
<a id=__codelineno-4-4 name=__codelineno-4-4 href=#__codelineno-4-4></a>SRR014335-chr1.fastq  SRR014336-chr1.fastq  SRR014337-chr1.fastq  SRR014339-chr1.fastq  SRR014340-chr1.fastq  SRR014341-chr1.fastq
</code></pre></div> <div class=highlight><pre><span></span><code><a id=__codelineno-5-1 name=__codelineno-5-1 href=#__codelineno-5-1></a><span class=k>for</span> filename <span class=k>in</span> *
<a id=__codelineno-5-2 name=__codelineno-5-2 href=#__codelineno-5-2></a><span class=k>do</span>
<a id=__codelineno-5-3 name=__codelineno-5-3 href=#__codelineno-5-3></a><span class=nv>base</span><span class=o>=</span><span class=k>$(</span>basename <span class=si>${</span><span class=nv>filename</span><span class=si>}</span> .trimmed.fastq<span class=k>)</span>
<a id=__codelineno-5-4 name=__codelineno-5-4 href=#__codelineno-5-4></a>hisat2 -p <span class=m>4</span> -x ../Genome/Saccharomyces_cerevisiae.R64-1-1.dna.toplevel -U <span class=nv>$filename</span> -S ../Mapping/<span class=si>${</span><span class=nv>base</span><span class=si>}</span>.sam --summary-file ../Mapping/<span class=si>${</span><span class=nv>base</span><span class=si>}</span>_summary.txt
<a id=__codelineno-5-5 name=__codelineno-5-5 href=#__codelineno-5-5></a><span class=k>done</span>
</code></pre></div></p> <p>Now we can explore our SAM files.</p> <div class=highlight><pre><span></span><code><a id=__codelineno-6-1 name=__codelineno-6-1 href=#__codelineno-6-1></a>$ <span class=nb>cd</span> ../Mapping
<a id=__codelineno-6-2 name=__codelineno-6-2 href=#__codelineno-6-2></a>
<a id=__codelineno-6-3 name=__codelineno-6-3 href=#__codelineno-6-3></a>$ ls
<a id=__codelineno-6-4 name=__codelineno-6-4 href=#__codelineno-6-4></a>SRR014335-chr1.sam          SRR014336-chr1_summary.txt  SRR014339-chr1.sam          SRR014340-chr1_summary.txt
<a id=__codelineno-6-5 name=__codelineno-6-5 href=#__codelineno-6-5></a>SRR014335-chr1_summary.txt  SRR014337-chr1.sam          SRR014339-chr1_summary.txt  SRR014341-chr1.sam
<a id=__codelineno-6-6 name=__codelineno-6-6 href=#__codelineno-6-6></a>SRR014336-chr1.sam          SRR014337-chr1_summary.txt  SRR014340-chr1.sam          SRR014341-chr1_summary.txt
</code></pre></div> <h2 id=converting-sam-files-to-bam-files>Converting SAM files to BAM files<a class=headerlink href=#converting-sam-files-to-bam-files title="Permanent link">&para;</a></h2> <p>The SAM file, is a tab-delimited text file that contains information for each individual read and its alignment to the genome. While we do not have time to go into detail about the features of the SAM format, the paper by Heng Li et al. provides a lot more detail on the specification.</p> <p>The compressed binary version of SAM is called a BAM file. We use this version to reduce size and to allow for indexing, which enables efficient random access of the data contained within the file.</p> <h3 id=a-quick-look-into-the-sam-file>A quick look into the sam file<a class=headerlink href=#a-quick-look-into-the-sam-file title="Permanent link">&para;</a></h3> <div class=highlight><pre><span></span><code><a id=__codelineno-7-1 name=__codelineno-7-1 href=#__codelineno-7-1></a>$ less SRR014335-chr1.sam 
<a id=__codelineno-7-2 name=__codelineno-7-2 href=#__codelineno-7-2></a>
<a id=__codelineno-7-3 name=__codelineno-7-3 href=#__codelineno-7-3></a>The file begins with a header, which is optional. The header is used to describe the <span class=nb>source</span> of data, reference sequence, method of alignment, etc., this will change depending on the aligner being used. Following the header is the alignment section. Each line that follows corresponds to alignment information <span class=k>for</span> a single read. Each alignment line has <span class=m>11</span> mandatory fields <span class=k>for</span> essential mapping information and a variable number of other fields <span class=k>for</span> aligner specific information. An example entry from a SAM file is displayed below with the different fields highlighted.
</code></pre></div> <p>We will convert the SAM file to BAM format using the samtools program with the <code>view</code> command </p> <p><div class=highlight><pre><span></span><code><a id=__codelineno-8-1 name=__codelineno-8-1 href=#__codelineno-8-1></a>$ module load SAMtools/1.10-GCC-9.2.0
</code></pre></div> <div class=highlight><pre><span></span><code><a id=__codelineno-9-1 name=__codelineno-9-1 href=#__codelineno-9-1></a><span class=k>for</span> filename <span class=k>in</span> *.sam
<a id=__codelineno-9-2 name=__codelineno-9-2 href=#__codelineno-9-2></a><span class=k>do</span>
<a id=__codelineno-9-3 name=__codelineno-9-3 href=#__codelineno-9-3></a><span class=nv>base</span><span class=o>=</span><span class=k>$(</span>basename <span class=si>${</span><span class=nv>filename</span><span class=si>}</span> .sam<span class=k>)</span>
<a id=__codelineno-9-4 name=__codelineno-9-4 href=#__codelineno-9-4></a>samtools view -S -b <span class=si>${</span><span class=nv>filename</span><span class=si>}</span> -o <span class=si>${</span><span class=nv>base</span><span class=si>}</span>.bam
<a id=__codelineno-9-5 name=__codelineno-9-5 href=#__codelineno-9-5></a><span class=k>done</span>
</code></pre></div></p> <div class="admonition quote"> <ul> <li><strong>-S</strong> input is in SAM format (it's here as a placeholder as SAMtoools version 1.10 or above will pick up the correct format automatically by examining the first few characters of input.)</li> <li><strong>-b</strong> Output is in BAM format</li> </ul> </div> <div class=highlight><pre><span></span><code><a id=__codelineno-10-1 name=__codelineno-10-1 href=#__codelineno-10-1></a>$ ls
<a id=__codelineno-10-2 name=__codelineno-10-2 href=#__codelineno-10-2></a>SRR014335-chr1.bam  SRR014336-chr1.bam  SRR014337-chr1.bam  SRR014339-chr1.bam  SRR014340-chr1.bam  SRR014341-chr1.bam
<a id=__codelineno-10-3 name=__codelineno-10-3 href=#__codelineno-10-3></a>SRR014335-chr1.sam  SRR014336-chr1.sam  SRR014337-chr1.sam  SRR014339-chr1.sam  SRR014340-chr1.sam  SRR014341-chr1.sam
</code></pre></div> <p>Next we sort the BAM file using the sort command from samtools. <code>-o</code> tells the command where to write the output.</p> <blockquote> <p><strong>SAM/BAM files can be sorted in multiple ways, e.g. by location of alignment on the chromosome, by read name, etc. It is important to be aware that different alignment tools will output differently sorted SAM/BAM, and different downstream tools require differently sorted alignment files as input.</strong></p> </blockquote> <div class=highlight><pre><span></span><code><a id=__codelineno-11-1 name=__codelineno-11-1 href=#__codelineno-11-1></a><span class=k>for</span> filename <span class=k>in</span> *.bam
<a id=__codelineno-11-2 name=__codelineno-11-2 href=#__codelineno-11-2></a><span class=k>do</span>
<a id=__codelineno-11-3 name=__codelineno-11-3 href=#__codelineno-11-3></a><span class=nv>base</span><span class=o>=</span><span class=k>$(</span>basename <span class=si>${</span><span class=nv>filename</span><span class=si>}</span> .bam<span class=k>)</span>
<a id=__codelineno-11-4 name=__codelineno-11-4 href=#__codelineno-11-4></a>samtools sort -o <span class=si>${</span><span class=nv>base</span><span class=si>}</span>_sorted.bam <span class=si>${</span><span class=nv>filename</span><span class=si>}</span>
<a id=__codelineno-11-5 name=__codelineno-11-5 href=#__codelineno-11-5></a><span class=k>done</span>
</code></pre></div> <p>You can use samtools to learn more about the bam file as well.</p> <h2 id=some-stats-on-your-mapping>Some stats on your mapping:<a class=headerlink href=#some-stats-on-your-mapping title="Permanent link">&para;</a></h2> <p><div class=highlight><pre><span></span><code><a id=__codelineno-12-1 name=__codelineno-12-1 href=#__codelineno-12-1></a>$ samtools flagstat SRR014335-chr1_sorted.bam 
</code></pre></div> <div class=highlight><pre><span></span><code><a id=__codelineno-13-1 name=__codelineno-13-1 href=#__codelineno-13-1></a>156984 + 0 in total (QC-passed reads + QC-failed reads)
<a id=__codelineno-13-2 name=__codelineno-13-2 href=#__codelineno-13-2></a>31894 + 0 secondary
<a id=__codelineno-13-3 name=__codelineno-13-3 href=#__codelineno-13-3></a>0 + 0 supplementary
<a id=__codelineno-13-4 name=__codelineno-13-4 href=#__codelineno-13-4></a>0 + 0 duplicates
<a id=__codelineno-13-5 name=__codelineno-13-5 href=#__codelineno-13-5></a>136447 + 0 mapped (86.92% : N/A)
<a id=__codelineno-13-6 name=__codelineno-13-6 href=#__codelineno-13-6></a>0 + 0 paired in sequencing
<a id=__codelineno-13-7 name=__codelineno-13-7 href=#__codelineno-13-7></a>0 + 0 read1
<a id=__codelineno-13-8 name=__codelineno-13-8 href=#__codelineno-13-8></a>0 + 0 read2
<a id=__codelineno-13-9 name=__codelineno-13-9 href=#__codelineno-13-9></a>0 + 0 properly paired (N/A : N/A)
<a id=__codelineno-13-10 name=__codelineno-13-10 href=#__codelineno-13-10></a>0 + 0 with itself and mate mapped
<a id=__codelineno-13-11 name=__codelineno-13-11 href=#__codelineno-13-11></a>0 + 0 singletons (N/A : N/A)
<a id=__codelineno-13-12 name=__codelineno-13-12 href=#__codelineno-13-12></a>0 + 0 with mate mapped to a different chr
<a id=__codelineno-13-13 name=__codelineno-13-13 href=#__codelineno-13-13></a>0 + 0 with mate mapped to a different chr (mapQ&gt;=5)
</code></pre></div> Basic statistics shown by <code>flagstat</code> will be slightly different from those in the summary file generated by HISAT2 due to different "totals" that are used for comparisons. <code>flagstat</code> compares the number of alignments; HISAT2 compares the number of reads mapped. This is because reads can be mapped/aligned to more than one reference location, and these reads have a "primary" and "secondary" alignment (see section 1.2 of the <a href=https://github.com/samtools/hts-specs/blob/master/SAMv1.pdf>SAM specifications</a>). For example, the percent overall alignment in the HISAT2 summary will be equivalent to the percent primary mapped evaluated by <code>flagstat</code>. To get number of reads that aligned 0 times (summary file), the equivalent statistic from <code>flagstat</code> would be subtracting the number of mapped reads from the number of total alignments.</p> <hr> <h2 id=multiqc-hisat2-output>MultiQC: <code>HISAT2</code> output<a class=headerlink href=#multiqc-hisat2-output title="Permanent link">&para;</a></h2> <ul> <li>The HISAT2 output data can also be incorporated into the MultiQC report the next time it is run.</li> </ul> <div class=highlight><pre><span></span><code><a id=__codelineno-14-1 name=__codelineno-14-1 href=#__codelineno-14-1></a> $ <span class=nb>cd</span> ~/RNA_seq/MultiQC
<a id=__codelineno-14-2 name=__codelineno-14-2 href=#__codelineno-14-2></a>
<a id=__codelineno-14-3 name=__codelineno-14-3 href=#__codelineno-14-3></a> $ cp ../Mapping/*summary* ./
<a id=__codelineno-14-4 name=__codelineno-14-4 href=#__codelineno-14-4></a>
<a id=__codelineno-14-5 name=__codelineno-14-5 href=#__codelineno-14-5></a> $ multiqc .
</code></pre></div> <p><img alt=image src=../Images/MQC3.png></p> <p>Please note: running HISAT2 with either option <code>--summary-file</code> or older versions (&lt; v2.1.0) gives log output identical to Bowtie2. These logs are indistinguishable and summary statistics will appear in MultiQC reports labelled as Bowtie2.</p> <hr> <h1 id=read-summarization>Read Summarization<a class=headerlink href=#read-summarization title="Permanent link">&para;</a></h1> <p>Sequencing reads often need to be assigned to genomic features of interest after they are mapped to the reference genome. This process is often called read summarization or read quantification. Read summarization is required by a number of downstream analyses such as gene expression analysis and histone modification analysis. The output of read summarization is a count table, in which the number of reads assigned to each feature in each library is recorded.</p> <h1 id=counting>Counting<a class=headerlink href=#counting title="Permanent link">&para;</a></h1> <ul> <li>We need to do some counting!</li> <li>Want to generate count data for each gene (actually each exon) - how many reads mapped to each exon in the genome, from each of our samples?</li> <li>Once we have that information, we can start thinking about how to determine which genes were differentially expressed in our study.</li> </ul> <h2 id=subread-and-featurecounts>Subread and FeatureCounts<a class=headerlink href=#subread-and-featurecounts title="Permanent link">&para;</a></h2> <ul> <li>The featureCounts tool from the Subread package can be used to count how many reads aligned to each genome feature (exon).</li> <li>Need to specify the annotation information (.gtf file) You can process all the samples at once:</li> </ul> <div class=highlight><pre><span></span><code><a id=__codelineno-15-1 name=__codelineno-15-1 href=#__codelineno-15-1></a>$ module load Subread/2.0.0-GCC-9.2.0
<a id=__codelineno-15-2 name=__codelineno-15-2 href=#__codelineno-15-2></a>
<a id=__codelineno-15-3 name=__codelineno-15-3 href=#__codelineno-15-3></a>$ <span class=nb>pwd</span>
<a id=__codelineno-15-4 name=__codelineno-15-4 href=#__codelineno-15-4></a>/home/<span class=nv>$USER</span>/RNA_seq
<a id=__codelineno-15-5 name=__codelineno-15-5 href=#__codelineno-15-5></a>
<a id=__codelineno-15-6 name=__codelineno-15-6 href=#__codelineno-15-6></a>$ mkdir Counts
<a id=__codelineno-15-7 name=__codelineno-15-7 href=#__codelineno-15-7></a>
<a id=__codelineno-15-8 name=__codelineno-15-8 href=#__codelineno-15-8></a>$ <span class=nb>cd</span> Counts
<a id=__codelineno-15-9 name=__codelineno-15-9 href=#__codelineno-15-9></a>
<a id=__codelineno-15-10 name=__codelineno-15-10 href=#__codelineno-15-10></a>$ featureCounts -a ../Genome/Saccharomyces_cerevisiae.R64-1-1.99.gtf -o ./yeast_counts.txt -T <span class=m>2</span> -t exon -g gene_id ../Mapping/*sorted.bam
</code></pre></div> <div class="admonition quote"> <ul> <li><strong>-a</strong> Name of an annotation file. GTF/GFF format by default</li> <li><strong>-o</strong> Name of output file including read counts</li> <li><strong>-T</strong> Specify the number of threads/CPUs used for mapping. 1 by default</li> <li><strong>-t</strong> Specify feature type in GTF annotation. 'exon' by default. Features used for read counting will be extracted from annotation using the provided value.</li> <li><strong>-g</strong> Specify attribute type in GTF annotation. 'gene_id' by default. Meta-features used for read counting will be extracted from annotation using the provided value.</li> </ul> </div> <hr> <h2 id=multiqc-featurecounts-output>MultiQC: featureCounts output<a class=headerlink href=#multiqc-featurecounts-output title="Permanent link">&para;</a></h2> <ul> <li>If the samples are processed individually, the output data can be incorporated into the MultiQC report the next time it is run.</li> </ul> <div class=highlight><pre><span></span><code><a id=__codelineno-16-1 name=__codelineno-16-1 href=#__codelineno-16-1></a>$ <span class=nb>cd</span> ../MultiQC
<a id=__codelineno-16-2 name=__codelineno-16-2 href=#__codelineno-16-2></a>
<a id=__codelineno-16-3 name=__codelineno-16-3 href=#__codelineno-16-3></a>$ cp ../Counts/* .
<a id=__codelineno-16-4 name=__codelineno-16-4 href=#__codelineno-16-4></a>
<a id=__codelineno-16-5 name=__codelineno-16-5 href=#__codelineno-16-5></a>$ multiqc .
</code></pre></div> <p><img alt=image src=../Images/MQC4.png></p> <hr> <p>Since we now have all the count data in one file, we need to transfer it to our local computers so we could start working on RStudio to get differentially expressed genes.</p> <ul> <li>And the code to do it is below, however for this workshop, we are going to use a different file (yeast_counts_all_chr.txt) that you can download from section 4. Differential_Expression. This file has all the data for all the chromosomes.</li> </ul> <div class=highlight><pre><span></span><code><a id=__codelineno-17-1 name=__codelineno-17-1 href=#__codelineno-17-1></a><span class=c1>#In a new terminal that you can access you computer files, cd to the directory you want to save the counts file.</span>
<a id=__codelineno-17-2 name=__codelineno-17-2 href=#__codelineno-17-2></a>
<a id=__codelineno-17-3 name=__codelineno-17-3 href=#__codelineno-17-3></a>$ scp fayfa80p@login.mahuika.nesi.org.nz:/home/fayfa80p/RNA_seq/Counts/yeast_counts_all_chr.txt ./
</code></pre></div> <hr> <p align=center><b><a class=btn href=https://genomicsaotearoa.github.io/RNA-seq-workshop/ style="background: var(--bs-dark);font-weight:bold">Back to homepage</a></b></p> </article> </div> </div> <a href=# class="md-top md-icon" data-md-component=top hidden> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg> Back to top </a> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../3_trimmingfiltering/ class="md-footer__link md-footer__link--prev" aria-label="Previous: Cleaning Reads" rel=prev> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Previous </span> Cleaning Reads </div> </div> </a> <a href=../5_rnaseq-diffexp/ class="md-footer__link md-footer__link--next" aria-label="Next: Differential Expression Analysis" rel=next> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Next </span> Differential Expression Analysis </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top"], "search": "../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script> <script src=../assets/javascripts/bundle.6c7ad80a.min.js></script> <script src=../javascripts/mathjax.js></script> <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script src=https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js></script> </body> </html>