
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa & NeSI">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/RNA-seq-workshop/5_rnaseq-diffexp/">
      
      <link rel="icon" href="../Images/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.3.0, mkdocs-material-8.3.9">
    
    
      
        <title>Differential Expression Analysis - RNA-Seq Data Analysis</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.1d29e8d0.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.cbb835fc.min.css">
        
      
      
    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CFira+Code:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Fira Code"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="light-green" data-md-color-accent="">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#differential-expression-analysis" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="RNA-Seq Data Analysis" class="md-header__button md-logo" aria-label="RNA-Seq Data Analysis" data-md-component="logo">
      
  <img src="../Images/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            RNA-Seq Data Analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Differential Expression Analysis
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="" data-md-color-primary="light-green" data-md-color-accent=""  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_3" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="light-green" data-md-color-accent=""  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_2">
          
            <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="light-green" data-md-color-accent=""  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_3">
          
            <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_2" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
            </label>
          
        
      </form>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/RNA-seq-workshop.git" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/RNA-seq-workshop
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="RNA-Seq Data Analysis" class="md-nav__button md-logo" aria-label="RNA-Seq Data Analysis" data-md-component="logo">
      
  <img src="../Images/nesi_ga.png" alt="logo">

    </a>
    RNA-Seq Data Analysis
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/RNA-seq-workshop.git" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/RNA-seq-workshop
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../1_background/" class="md-nav__link">
        RNA-seq Workflow
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../2_qualityassessment/" class="md-nav__link">
        Quality control of the sequencing data.
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../3_trimmingfiltering/" class="md-nav__link">
        Cleaning Reads
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../4_mappingANDcount/" class="md-nav__link">
        Mapping and Counting
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Differential Expression Analysis
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Differential Expression Analysis
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#recap" class="md-nav__link">
    Recap
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-set-reminder" class="md-nav__link">
    Data set reminder
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-organised" class="md-nav__link">
    Getting organised
  </a>
  
    <nav class="md-nav" aria-label="Getting organised">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#create-an-rstudio-project" class="md-nav__link">
    Create an RStudio project
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#count-data" class="md-nav__link">
    Count data
  </a>
  
    <nav class="md-nav" aria-label="Count data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#visualising-the-data" class="md-nav__link">
    Visualising the data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#read-counts-per-sample" class="md-nav__link">
    Read counts per sample
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#detecting-differential-expression" class="md-nav__link">
    Detecting differential expression:
  </a>
  
    <nav class="md-nav" aria-label="Detecting differential expression:">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#limma" class="md-nav__link">
    Limma
  </a>
  
    <nav class="md-nav" aria-label="Limma">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#aside-rpkm" class="md-nav__link">
    Aside: RPKM
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#back-to-the-analysis-using-logcpm" class="md-nav__link">
    Back to the analysis… (using logCPM)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#limma-voom" class="md-nav__link">
    Limma: voom
  </a>
  
    <nav class="md-nav" aria-label="Limma: voom">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#voom-impact-on-samples" class="md-nav__link">
    Voom: impact on samples
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#statistical-testing-fitting-a-linear-model" class="md-nav__link">
    Statistical testing: fitting a linear model
  </a>
  
    <nav class="md-nav" aria-label="Statistical testing: fitting a linear model">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#significant-genes" class="md-nav__link">
    Significant genes
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deseq2" class="md-nav__link">
    DESeq2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#edger" class="md-nav__link">
    edgeR
  </a>
  
    <nav class="md-nav" aria-label="edgeR">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#identify-degs-with-edgers-exact-method" class="md-nav__link">
    Identify DEGs with edgeR’s “Exact” Method
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#edger-analysis-and-output" class="md-nav__link">
    edgeR analysis and output
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deseq2-vs-edger" class="md-nav__link">
    DESeq2 vs edgeR
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#limma-vs-edger-vs-deseq2" class="md-nav__link">
    Limma vs edgeR vs DESeq2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#summary" class="md-nav__link">
    Summary
  </a>
  
    <nav class="md-nav" aria-label="Summary">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#save-limma-toptable-results-for-next-session" class="md-nav__link">
    Save limma topTable results for next session…
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../6_over-representation_analysis/" class="md-nav__link">
        Over-representation analysis (Gene Ontology)
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../supp_1/" class="md-nav__link">
        A Guide to RNA-Seq
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../supp_2/" class="md-nav__link">
        Lecture notes : Introduction To Differential Expression Analysis
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../supp_3/" class="md-nav__link">
        Lecture notes : Annotation Data & Gene Set Analysis
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#recap" class="md-nav__link">
    Recap
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#data-set-reminder" class="md-nav__link">
    Data set reminder
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getting-organised" class="md-nav__link">
    Getting organised
  </a>
  
    <nav class="md-nav" aria-label="Getting organised">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#create-an-rstudio-project" class="md-nav__link">
    Create an RStudio project
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#count-data" class="md-nav__link">
    Count data
  </a>
  
    <nav class="md-nav" aria-label="Count data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#visualising-the-data" class="md-nav__link">
    Visualising the data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#read-counts-per-sample" class="md-nav__link">
    Read counts per sample
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#detecting-differential-expression" class="md-nav__link">
    Detecting differential expression:
  </a>
  
    <nav class="md-nav" aria-label="Detecting differential expression:">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#limma" class="md-nav__link">
    Limma
  </a>
  
    <nav class="md-nav" aria-label="Limma">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#aside-rpkm" class="md-nav__link">
    Aside: RPKM
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#back-to-the-analysis-using-logcpm" class="md-nav__link">
    Back to the analysis… (using logCPM)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#limma-voom" class="md-nav__link">
    Limma: voom
  </a>
  
    <nav class="md-nav" aria-label="Limma: voom">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#voom-impact-on-samples" class="md-nav__link">
    Voom: impact on samples
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#statistical-testing-fitting-a-linear-model" class="md-nav__link">
    Statistical testing: fitting a linear model
  </a>
  
    <nav class="md-nav" aria-label="Statistical testing: fitting a linear model">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#significant-genes" class="md-nav__link">
    Significant genes
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deseq2" class="md-nav__link">
    DESeq2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#edger" class="md-nav__link">
    edgeR
  </a>
  
    <nav class="md-nav" aria-label="edgeR">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#identify-degs-with-edgers-exact-method" class="md-nav__link">
    Identify DEGs with edgeR’s “Exact” Method
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#edger-analysis-and-output" class="md-nav__link">
    edgeR analysis and output
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#deseq2-vs-edger" class="md-nav__link">
    DESeq2 vs edgeR
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#limma-vs-edger-vs-deseq2" class="md-nav__link">
    Limma vs edgeR vs DESeq2
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#summary" class="md-nav__link">
    Summary
  </a>
  
    <nav class="md-nav" aria-label="Summary">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#save-limma-toptable-results-for-next-session" class="md-nav__link">
    Save limma topTable results for next session…
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
  <a href="https://github.com/GenomicsAotearoa/RNA-seq-workshop.git/edit/master/docs/5_rnaseq-diffexp.md" title="Edit this page" class="md-content__button md-icon">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
  </a>



<h1 id="differential-expression-analysis">Differential Expression Analysis<a class="headerlink" href="#differential-expression-analysis" title="Permanent link">&para;</a></h1>
<h3 id="recap">Recap<a class="headerlink" href="#recap" title="Permanent link">&para;</a></h3>
<ul>
<li>In the last section we worked through the process of quality
    assessment and alignment for RNA-seq data</li>
<li>This typically takes place on the command line, but can also be done
    from within R.</li>
<li>The end result was the generation of count data (counts of reads
    aligned to each gene, per sample) using the FeatureCounts command
    from Subread/Rsubread.</li>
<li>Now that we’ve got count data in R, we can begin our differential
    expression analysis.</li>
</ul>
<h3 id="data-set-reminder">Data set reminder<a class="headerlink" href="#data-set-reminder" title="Permanent link">&para;</a></h3>
<ul>
<li>Data obtained from yeast RNA-seq experiment,
    <a href="https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000299">Lee
    et al 2008 </a></li>
<li>Wild-type versus RNA degradation mutants</li>
<li>Six samples (3 WT / 3 MT)</li>
<li>
<p>NOTE: since we are dealing with an RNA degradation mutant yeast
    strain, we will see MAJOR transcriptomic differences between the
    wild-type and mutant groups (usually the differences you observe in
    a typical RNA-seq study wouldn’t be this extreme).</p>
</li>
<li>
<p>Slides: <a href="https://github.com/GenomicsAotearoa/RNA-seq-workshop/blob/master/4.Differential_Expression/lecture_differential_expression.pdf">lecture_differential_expression.pdf</a></p>
</li>
</ul>
<h2 id="getting-organised">Getting organised<a class="headerlink" href="#getting-organised" title="Permanent link">&para;</a></h2>
<h3 id="create-an-rstudio-project">Create an RStudio project<a class="headerlink" href="#create-an-rstudio-project" title="Permanent link">&para;</a></h3>
<p><strong>NOTE: skip to the next section (“Count Data”) if you are working
within Jupyter on NeSI</strong></p>
<p>One of the first benefits we will take advantage of in RStudio is
something called an RStudio Project. An RStudio project allows you to
more easily:</p>
<ul>
<li>Save data, files, variables, packages, etc. related to a specific
    analysis project</li>
<li>Restart work where you left off</li>
<li>Collaborate, especially if you are using version control such as
    git.</li>
</ul>
<p>To create a project:</p>
<ul>
<li>Open RStudio and go to the File menu, and click New Project.</li>
<li>In the window that opens select Existing Project, and browse to the
    RNA_seq folder.</li>
<li>Finally, click Create Project.</li>
</ul>
<p><em>Save source from untitled to <code>yeast_data.R</code> and continue saving
regularly as you work.</em></p>
<h3 id="count-data">Count data<a class="headerlink" href="#count-data" title="Permanent link">&para;</a></h3>
<p>Note: I have now aligned the data for ALL CHROMOSOMES and generated
counts, so we are working with data from all 7127 genes.</p>
<p><em>Let’s look at our data set and perform some basic checks before we do a
differential expression analysis.</em></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="n">fcData</span> <span class="o">=</span> <span class="nf">read.table</span><span class="p">(</span><span class="s">&#39;yeast_counts_all_chr.txt&#39;</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s">&#39;\t&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a><span class="n">fcData</span> <span class="o">%&gt;%</span> <span class="nf">head</span><span class="p">()</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##      Geneid Chr Start   End Strand Length ...STAR.SRR014335.Aligned.out.sam
## 1   YDL248W  IV  1802  2953      +   1152                                52
## 2 YDL247W-A  IV  3762  3836      +     75                                 0
## 3   YDL247W  IV  5985  7814      +   1830                                 2
## 4   YDL246C  IV  8683  9756      -   1074                                 0
## 5   YDL245C  IV 11657 13360      -   1704                                 0
## 6   YDL244W  IV 16204 17226      +   1023                                 6
##   ...STAR.SRR014336.Aligned.out.sam ...STAR.SRR014337.Aligned.out.sam
## 1                                46                                36
## 2                                 0                                 0
## 3                                 4                                 2
## 4                                 0                                 1
## 5                                 3                                 0
## 6                                 6                                 5
##   ...STAR.SRR014339.Aligned.out.sam ...STAR.SRR014340.Aligned.out.sam
## 1                                65                                70
## 2                                 0                                 1
## 3                                 6                                 8
## 4                                 1                                 2
## 5                                 5                                 7
## 6                                20                                30
##   ...STAR.SRR014341.Aligned.out.sam
## 1                                78
## 2                                 0
## 3                                 5
## 4                                 0
## 5                                 4
## 6                                19
</code></pre></div>
<p>Check dimensions:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="nf">dim</span><span class="p">(</span><span class="n">fcData</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 7127   12
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="nf">names</span><span class="p">(</span><span class="n">fcData</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##  [1] &quot;Geneid&quot;                            &quot;Chr&quot;                              
##  [3] &quot;Start&quot;                             &quot;End&quot;                              
##  [5] &quot;Strand&quot;                            &quot;Length&quot;                           
##  [7] &quot;...STAR.SRR014335.Aligned.out.sam&quot; &quot;...STAR.SRR014336.Aligned.out.sam&quot;
##  [9] &quot;...STAR.SRR014337.Aligned.out.sam&quot; &quot;...STAR.SRR014339.Aligned.out.sam&quot;
## [11] &quot;...STAR.SRR014340.Aligned.out.sam&quot; &quot;...STAR.SRR014341.Aligned.out.sam&quot;
</code></pre></div>
<p>Rename data columns to reflect group membership</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="nf">names</span><span class="p">(</span><span class="n">fcData</span><span class="p">)[</span><span class="m">7</span><span class="o">:</span><span class="m">12</span><span class="p">]</span> <span class="o">=</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;WT1&quot;</span><span class="p">,</span> <span class="s">&quot;WT2&quot;</span><span class="p">,</span> <span class="s">&quot;WT3&quot;</span><span class="p">,</span> <span class="s">&quot;MT1&quot;</span><span class="p">,</span> <span class="s">&quot;MT2&quot;</span><span class="p">,</span> <span class="s">&quot;MT3&quot;</span><span class="p">)</span>
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>
<a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a><span class="n">fcData</span> <span class="o">%&gt;%</span> <span class="nf">head</span><span class="p">()</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##      Geneid Chr Start   End Strand Length WT1 WT2 WT3 MT1 MT2 MT3
## 1   YDL248W  IV  1802  2953      +   1152  52  46  36  65  70  78
## 2 YDL247W-A  IV  3762  3836      +     75   0   0   0   0   1   0
## 3   YDL247W  IV  5985  7814      +   1830   2   4   2   6   8   5
## 4   YDL246C  IV  8683  9756      -   1074   0   0   1   1   2   0
## 5   YDL245C  IV 11657 13360      -   1704   0   3   0   5   7   4
## 6   YDL244W  IV 16204 17226      +   1023   6   6   5  20  30  19
</code></pre></div>
<p>Extract count data</p>
<ul>
<li>Remove annotation columns</li>
<li>Add row names</li>
</ul>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="n">counts</span> <span class="o">=</span> <span class="n">fcData</span><span class="p">[,</span> <span class="m">7</span><span class="o">:</span><span class="m">12</span><span class="p">]</span>
<a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a><span class="nf">rownames</span><span class="p">(</span><span class="n">counts</span><span class="p">)</span> <span class="o">=</span> <span class="n">fcData</span><span class="o">$</span><span class="n">Geneid</span>
<a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a><span class="n">counts</span> <span class="o">%&gt;%</span> <span class="nf">head</span><span class="p">()</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##           WT1 WT2 WT3 MT1 MT2 MT3
## YDL248W    52  46  36  65  70  78
## YDL247W-A   0   0   0   0   1   0
## YDL247W     2   4   2   6   8   5
## YDL246C     0   0   1   1   2   0
## YDL245C     0   3   0   5   7   4
## YDL244W     6   6   5  20  30  19
</code></pre></div>
<h4 id="visualising-the-data">Visualising the data<a class="headerlink" href="#visualising-the-data" title="Permanent link">&para;</a></h4>
<p>Data are highly skewed (suggests that logging might be useful):</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="nf">boxplot</span><span class="p">(</span><span class="nf">as.matrix</span><span class="p">(</span><span class="n">counts</span><span class="p">)</span> <span class="o">~</span> <span class="nf">col</span><span class="p">(</span><span class="n">counts</span><span class="p">))</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-5-1.png" /><!-- --></p>
<p>Some genes have zero counts:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="nf">colSums</span><span class="p">(</span><span class="n">counts</span><span class="o">==</span><span class="m">0</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## WT1 WT2 WT3 MT1 MT2 MT3 
## 562 563 573 437 425 435
</code></pre></div>
<p>Log transformation (add 0.5 to avoid log(0) issues):</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="n">logCounts</span> <span class="o">=</span> <span class="nf">log2</span><span class="p">(</span><span class="nf">as.matrix</span><span class="p">(</span><span class="n">counts</span><span class="p">)</span><span class="o">+</span> <span class="m">0.5</span><span class="p">)</span>
</code></pre></div>
<p>Now we can see the per-sample distributions more clearly:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="nf">boxplot</span><span class="p">(</span><span class="nf">as.matrix</span><span class="p">(</span><span class="n">logCounts</span><span class="p">)</span> <span class="o">~</span> <span class="nf">col</span><span class="p">(</span><span class="n">logCounts</span><span class="p">))</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-8-1.png" /><!-- --></p>
<p>Density plots are also a good way to visualise the data:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="n">lineColour</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;blue&quot;</span><span class="p">,</span> <span class="s">&quot;blue&quot;</span><span class="p">,</span> <span class="s">&quot;blue&quot;</span><span class="p">,</span> <span class="s">&quot;red&quot;</span><span class="p">,</span> <span class="s">&quot;red&quot;</span><span class="p">,</span> <span class="s">&quot;red&quot;</span><span class="p">)</span>
<a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a><span class="n">lineColour</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] &quot;blue&quot; &quot;blue&quot; &quot;blue&quot; &quot;red&quot;  &quot;red&quot;  &quot;red&quot;
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="nf">plot</span><span class="p">(</span><span class="nf">density</span><span class="p">(</span><span class="n">logCounts</span><span class="p">[,</span><span class="m">1</span><span class="p">]),</span> <span class="n">ylim</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0.3</span><span class="p">),</span> <span class="n">col</span><span class="o">=</span><span class="n">lineColour</span><span class="p">[</span><span class="m">1</span><span class="p">])</span>
<a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a><span class="nf">for</span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">2</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">logCounts</span><span class="p">))</span> <span class="nf">lines</span><span class="p">(</span><span class="nf">density</span><span class="p">(</span><span class="n">logCounts</span><span class="p">[,</span><span class="n">i</span><span class="p">]),</span> <span class="n">col</span><span class="o">=</span><span class="n">lineColour</span><span class="p">[</span><span class="n">i</span><span class="p">])</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-9-1.png" /><!-- --></p>
<p>The boxplots and density plots show clear differences between the sample
groups - are these biological, or experimental artifacts? (often we
don’t know).</p>
<p><strong>Remember: the wild-type and mutant yeast strains are VERY different.
You wouldn’t normally see this amount of difference in the distributions
between the groups.</strong></p>
<h4 id="read-counts-per-sample">Read counts per sample<a class="headerlink" href="#read-counts-per-sample" title="Permanent link">&para;</a></h4>
<p>Normalisation process (slightly different for each analysis method)
takes “library size” (number of reads generated for each sample) into
account.</p>
<ul>
<li>A good blog about normalization:
    <a href="https://www.reneshbedre.com/blog/expression_units.html">Link</a></li>
</ul>
<div class="highlight"><pre><span></span><code><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="nf">colSums</span><span class="p">(</span><span class="n">counts</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##     WT1     WT2     WT3     MT1     MT2     MT3 
## 4915975 4892227 4778158 4618409 4719413 4554283
</code></pre></div>
<p>Visualise via bar plot</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="nf">colSums</span><span class="p">(</span><span class="n">counts</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">barplot</span><span class="p">(</span><span class="n">.</span><span class="p">,</span> <span class="n">ylab</span><span class="o">=</span><span class="s">&quot;Reads mapped per sample&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-11-1.png" /><!-- --></p>
<p><em>Now we are ready for differential expression analysis</em></p>
<h2 id="detecting-differential-expression">Detecting differential expression:<a class="headerlink" href="#detecting-differential-expression" title="Permanent link">&para;</a></h2>
<p>We are going to identify genes that are differential expressed using 3
different packages (time allowing) and compare the results.</p>
<p>But first, lets take a brief aside, and talk about the process of
detecting genes that have undergone a statistically significance change
in expression between the two groups.</p>
<p><strong>GO BACK TO THE README PAGE, AND OPEN UP THE PDF DOCUMENT</strong></p>
<hr />
<h3 id="limma">Limma<a class="headerlink" href="#limma" title="Permanent link">&para;</a></h3>
<!-- NEED TO EXPLAIN "TREND=TRUE" -->

<ul>
<li>Limma can be used for analysis, by transforming the RNA-seq count
    data in an appropriate way (log-scale normality-based assumption
    rather than Negative Binomial for count data)</li>
<li>Use data transformation and log to satisfy normality assumptions
    (CPM = Counts per Million).</li>
</ul>
<div class="highlight"><pre><span></span><code><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">limma</span><span class="p">)</span>
<a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a><span class="nf">library</span><span class="p">(</span><span class="n">edgeR</span><span class="p">)</span>
<a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a>
<a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a><span class="n">dge</span> <span class="o">&lt;-</span> <span class="nf">DGEList</span><span class="p">(</span><span class="n">counts</span><span class="o">=</span><span class="n">counts</span><span class="p">)</span>
<a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a><span class="n">dge</span> <span class="o">&lt;-</span> <span class="nf">calcNormFactors</span><span class="p">(</span><span class="n">dge</span><span class="p">)</span>
<a id="__codelineno-13-6" name="__codelineno-13-6" href="#__codelineno-13-6"></a><span class="n">logCPM</span> <span class="o">&lt;-</span> <span class="nf">cpm</span><span class="p">(</span><span class="n">dge</span><span class="p">,</span> <span class="n">log</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> <span class="n">prior.count</span><span class="o">=</span><span class="m">3</span><span class="p">)</span>
<a id="__codelineno-13-7" name="__codelineno-13-7" href="#__codelineno-13-7"></a>
<a id="__codelineno-13-8" name="__codelineno-13-8" href="#__codelineno-13-8"></a><span class="nf">options</span><span class="p">(</span><span class="n">width</span><span class="o">=</span><span class="m">100</span><span class="p">)</span>
<a id="__codelineno-13-9" name="__codelineno-13-9" href="#__codelineno-13-9"></a><span class="nf">head</span><span class="p">(</span><span class="n">logCPM</span><span class="p">,</span> <span class="m">3</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##                  WT1        WT2        WT3        MT1        MT2        MT3
## YDL248W    3.7199528  3.5561232  3.2538405  3.6446399  3.7156488  3.9155366
## YDL247W-A -0.6765789 -0.6765789 -0.6765789 -0.6765789 -0.3140297 -0.6765789
## YDL247W    0.1484688  0.6727144  0.1645731  0.7843936  1.0395626  0.6349276
</code></pre></div>
<h4 id="aside-rpkm">Aside: RPKM<a class="headerlink" href="#aside-rpkm" title="Permanent link">&para;</a></h4>
<p>We can use R to generate RPKM values (or FPKM if using paired-end
reads):</p>
<ul>
<li>Need gene length information to do this</li>
<li>Can get this from<code>goseq</code> package (we’ll use this later for our
    pathway analysis)</li>
</ul>
<div class="highlight"><pre><span></span><code><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">goseq</span><span class="p">)</span>
<a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a><span class="n">geneLengths</span> <span class="o">=</span> <span class="nf">getlength</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">counts</span><span class="p">),</span> <span class="s">&quot;sacCer2&quot;</span><span class="p">,</span><span class="s">&quot;ensGene&quot;</span><span class="p">)</span>
<a id="__codelineno-14-3" name="__codelineno-14-3" href="#__codelineno-14-3"></a><span class="n">rpkmData</span> <span class="o">&lt;-</span> <span class="nf">rpkm</span><span class="p">(</span><span class="n">dge</span><span class="p">,</span> <span class="n">geneLengths</span><span class="p">)</span>
<a id="__codelineno-14-4" name="__codelineno-14-4" href="#__codelineno-14-4"></a><span class="n">rpkmData</span> <span class="o">%&gt;%</span> <span class="nf">round</span><span class="p">(</span><span class="n">.</span><span class="p">,</span> <span class="m">2</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">head</span><span class="p">()</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##             WT1  WT2  WT3   MT1   MT2   MT3
## YDL248W   10.89 9.66 7.73 10.30 10.85 12.55
## YDL247W-A  0.00 0.00 0.00  0.00  2.35  0.00
## YDL247W    0.26 0.53 0.27  0.60  0.78  0.51
## YDL246C    0.00 0.00 0.23  0.17  0.33  0.00
## YDL245C    0.00 0.43 0.00  0.54  0.73  0.44
## YDL244W    1.41 1.42 1.21  3.57  5.24  3.44
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="nf">range</span><span class="p">(</span><span class="n">rpkmData</span><span class="p">,</span> <span class="n">na.rm</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1]     0.00 69786.58
</code></pre></div>
<p>Compare RPKM to logCPM</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a><span class="nf">par</span><span class="p">(</span><span class="n">mfrow</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">3</span><span class="p">))</span>
<a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a><span class="nf">for</span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">rpkmData</span><span class="p">)){</span>
<a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a>  <span class="nf">plot</span><span class="p">(</span><span class="n">logCPM</span><span class="p">[,</span><span class="n">i</span><span class="p">],</span> <span class="nf">log2</span><span class="p">(</span><span class="n">rpkmData</span><span class="p">[,</span><span class="n">i</span><span class="p">]),</span> <span class="n">pch</span><span class="o">=</span><span class="s">&#39;.&#39;</span><span class="p">,</span> 
<a id="__codelineno-16-4" name="__codelineno-16-4" href="#__codelineno-16-4"></a>       <span class="n">xlab</span><span class="o">=</span><span class="s">&quot;logCPM&quot;</span><span class="p">,</span> <span class="n">ylab</span><span class="o">=</span><span class="s">&quot;RPKM&quot;</span><span class="p">,</span> <span class="n">main</span><span class="o">=</span><span class="nf">colnames</span><span class="p">(</span><span class="n">logCPM</span><span class="p">)[</span><span class="n">i</span><span class="p">])</span>
<a id="__codelineno-16-5" name="__codelineno-16-5" href="#__codelineno-16-5"></a><span class="p">}</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-14-1.png" /><!-- --></p>
<p><strong>We’re NOT going to use RPKM data here. I just wanted to show you how
to calculate it, and how it relates to the logCPM data</strong></p>
<h3 id="back-to-the-analysis-using-logcpm">Back to the analysis… (using logCPM)<a class="headerlink" href="#back-to-the-analysis-using-logcpm" title="Permanent link">&para;</a></h3>
<p>What if we just did a t-test?</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a><span class="c1">## The beeswarm package is great for making jittered dot plots</span>
<a id="__codelineno-17-2" name="__codelineno-17-2" href="#__codelineno-17-2"></a><span class="nf">library</span><span class="p">(</span><span class="n">beeswarm</span><span class="p">)</span>
<a id="__codelineno-17-3" name="__codelineno-17-3" href="#__codelineno-17-3"></a>
<a id="__codelineno-17-4" name="__codelineno-17-4" href="#__codelineno-17-4"></a><span class="c1"># Specify &quot;conditions&quot; (groups: WT and MT)</span>
<a id="__codelineno-17-5" name="__codelineno-17-5" href="#__codelineno-17-5"></a><span class="n">conds</span> <span class="o">&lt;-</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;WT&quot;</span><span class="p">,</span><span class="s">&quot;WT&quot;</span><span class="p">,</span><span class="s">&quot;WT&quot;</span><span class="p">,</span><span class="s">&quot;MT&quot;</span><span class="p">,</span><span class="s">&quot;MT&quot;</span><span class="p">,</span><span class="s">&quot;MT&quot;</span><span class="p">)</span>
<a id="__codelineno-17-6" name="__codelineno-17-6" href="#__codelineno-17-6"></a>
<a id="__codelineno-17-7" name="__codelineno-17-7" href="#__codelineno-17-7"></a><span class="c1">## Perform t-test for &quot;gene number 6&quot; (because I like that one...)</span>
<a id="__codelineno-17-8" name="__codelineno-17-8" href="#__codelineno-17-8"></a><span class="nf">t.test</span><span class="p">(</span><span class="n">logCPM</span><span class="p">[</span><span class="m">6</span><span class="p">,]</span> <span class="o">~</span> <span class="n">conds</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## 
##  Welch Two Sample t-test
## 
## data:  logCPM[6, ] by conds
## t = 7.0124, df = 2.3726, p-value = 0.01228
## alternative hypothesis: true difference in means is not equal to 0
## 95 percent confidence interval:
##  0.5840587 1.9006761
## sample estimates:
## mean in group MT mean in group WT 
##         2.244310         1.001943
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a><span class="nf">beeswarm</span><span class="p">(</span><span class="n">logCPM</span><span class="p">[</span><span class="m">6</span><span class="p">,]</span> <span class="o">~</span> <span class="n">conds</span><span class="p">,</span> <span class="n">pch</span><span class="o">=</span><span class="m">16</span><span class="p">,</span> <span class="n">ylab</span><span class="o">=</span><span class="s">&quot;Expression (logCPM)&quot;</span><span class="p">,</span> <span class="n">xlab</span><span class="o">=</span><span class="s">&quot;Group&quot;</span><span class="p">,</span>
<a id="__codelineno-18-2" name="__codelineno-18-2" href="#__codelineno-18-2"></a>         <span class="n">main</span><span class="o">=</span><span class="nf">paste0</span><span class="p">(</span><span class="nf">rownames</span><span class="p">(</span><span class="n">logCPM</span><span class="p">)[</span><span class="m">6</span><span class="p">],</span> <span class="s">&quot;: MT vs WT&quot;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-15-1.png" /><!-- --></p>
<ul>
<li>we basically want to do this sort of analysis, for every gene</li>
<li>we’ll use a slightly more sophisticated approach though</li>
</ul>
<p>However, before we get to statistical testing, we (might) first need to
do a little bit more normalisation.</p>
<h3 id="limma-voom">Limma: voom<a class="headerlink" href="#limma-voom" title="Permanent link">&para;</a></h3>
<ul>
<li>The “voom” function estimates relationship between the mean and the
    variance of the logCPM data, normalises the data, and creates
    “precision weights” for each observation that are incorporated into
    the limma analysis.</li>
</ul>
<div class="highlight"><pre><span></span><code><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a><span class="n">design</span> <span class="o">&lt;-</span> <span class="nf">model.matrix</span><span class="p">(</span><span class="o">~</span><span class="n">conds</span><span class="p">)</span>
<a id="__codelineno-19-2" name="__codelineno-19-2" href="#__codelineno-19-2"></a><span class="n">design</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##   (Intercept) condsWT
## 1           1       1
## 2           1       1
## 3           1       1
## 4           1       0
## 5           1       0
## 6           1       0
## attr(,&quot;assign&quot;)
## [1] 0 1
## attr(,&quot;contrasts&quot;)
## attr(,&quot;contrasts&quot;)$conds
## [1] &quot;contr.treatment&quot;
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a><span class="n">v</span> <span class="o">&lt;-</span> <span class="nf">voom</span><span class="p">(</span><span class="n">dge</span><span class="p">,</span> <span class="n">design</span><span class="p">,</span> <span class="n">plot</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-16-1.png" /><!-- --></p>
<!-- ![Alt text](https://github.com/foreal17/RNA-seq-workshop/blob/master/Prep_Files/Images/Voom_Mean_Variance.png) -->

<h4 id="voom-impact-on-samples">Voom: impact on samples<a class="headerlink" href="#voom-impact-on-samples" title="Permanent link">&para;</a></h4>
<p>Mainly affects the low-end (low abundance genes)</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a><span class="nf">par</span><span class="p">(</span><span class="n">mfrow</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">3</span><span class="p">))</span>
<a id="__codelineno-21-2" name="__codelineno-21-2" href="#__codelineno-21-2"></a><span class="nf">for</span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">logCPM</span><span class="p">)){</span>
<a id="__codelineno-21-3" name="__codelineno-21-3" href="#__codelineno-21-3"></a>  <span class="nf">plot</span><span class="p">(</span><span class="n">logCPM</span><span class="p">[,</span><span class="n">i</span><span class="p">],</span> <span class="n">v</span><span class="o">$</span><span class="n">E</span><span class="p">[,</span><span class="n">i</span><span class="p">],</span> <span class="n">xlab</span><span class="o">=</span><span class="s">&quot;LogCPM&quot;</span><span class="p">,</span>
<a id="__codelineno-21-4" name="__codelineno-21-4" href="#__codelineno-21-4"></a>       <span class="n">ylab</span><span class="o">=</span><span class="s">&quot;Voom&quot;</span><span class="p">,</span><span class="n">main</span><span class="o">=</span><span class="nf">colnames</span><span class="p">(</span><span class="n">logCPM</span><span class="p">)[</span><span class="n">i</span><span class="p">])</span>
<a id="__codelineno-21-5" name="__codelineno-21-5" href="#__codelineno-21-5"></a>  <span class="nf">abline</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">1</span><span class="p">)</span>
<a id="__codelineno-21-6" name="__codelineno-21-6" href="#__codelineno-21-6"></a><span class="p">}</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-17-1.png" /><!-- --></p>
<p>Hasn’t removed the differences between the groups</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-22-1" name="__codelineno-22-1" href="#__codelineno-22-1"></a><span class="nf">boxplot</span><span class="p">(</span><span class="n">v</span><span class="o">$</span><span class="n">E</span> <span class="o">~</span> <span class="nf">col</span><span class="p">(</span><span class="n">v</span><span class="o">$</span><span class="n">E</span><span class="p">))</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-18-1.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-23-1" name="__codelineno-23-1" href="#__codelineno-23-1"></a><span class="n">lineColour</span> <span class="o">&lt;-</span> <span class="nf">ifelse</span><span class="p">(</span><span class="n">conds</span><span class="o">==</span><span class="s">&quot;MT&quot;</span><span class="p">,</span> <span class="s">&quot;red&quot;</span><span class="p">,</span> <span class="s">&quot;blue&quot;</span><span class="p">)</span>
<a id="__codelineno-23-2" name="__codelineno-23-2" href="#__codelineno-23-2"></a><span class="n">lineColour</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] &quot;blue&quot; &quot;blue&quot; &quot;blue&quot; &quot;red&quot;  &quot;red&quot;  &quot;red&quot;
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-24-1" name="__codelineno-24-1" href="#__codelineno-24-1"></a><span class="nf">plot</span><span class="p">(</span><span class="nf">density</span><span class="p">(</span><span class="n">v</span><span class="o">$</span><span class="n">E</span><span class="p">[,</span><span class="m">1</span><span class="p">]),</span> <span class="n">ylim</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0.3</span><span class="p">),</span> <span class="n">col</span><span class="o">=</span><span class="n">lineColour</span><span class="p">[</span><span class="m">1</span><span class="p">])</span>
<a id="__codelineno-24-2" name="__codelineno-24-2" href="#__codelineno-24-2"></a><span class="nf">for</span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">2</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">logCPM</span><span class="p">))</span> <span class="nf">lines</span><span class="p">(</span><span class="nf">density</span><span class="p">(</span><span class="n">v</span><span class="o">$</span><span class="n">E</span><span class="p">[,</span><span class="n">i</span><span class="p">]),</span> <span class="n">col</span><span class="o">=</span><span class="n">lineColour</span><span class="p">[</span><span class="n">i</span><span class="p">])</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-19-1.png" /><!-- --></p>
<p>We can deal with this via “quantile normalisation”. Specify</p>
<div class="highlight"><pre><span></span><code>normalize=&quot;quantile&quot;
</code></pre></div>
<p>in the <code>voom</code> function.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-25-1" name="__codelineno-25-1" href="#__codelineno-25-1"></a><span class="n">q</span> <span class="o">&lt;-</span> <span class="nf">voom</span><span class="p">(</span><span class="n">dge</span><span class="p">,</span> <span class="n">design</span><span class="p">,</span> <span class="n">plot</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> <span class="n">normalize</span><span class="o">=</span><span class="s">&quot;quantile&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-20-1.png" /><!-- --></p>
<p>Quantile normalisation forces all of the distributions to be the same.</p>
<ul>
<li>it is a strong assumption, and can have a major impact on the
    analysis</li>
<li>potentially removing biological differences between groups.</li>
</ul>
<div class="highlight"><pre><span></span><code><a id="__codelineno-26-1" name="__codelineno-26-1" href="#__codelineno-26-1"></a><span class="nf">par</span><span class="p">(</span><span class="n">mfrow</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">3</span><span class="p">))</span>
<a id="__codelineno-26-2" name="__codelineno-26-2" href="#__codelineno-26-2"></a><span class="nf">for</span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">logCPM</span><span class="p">)){</span>
<a id="__codelineno-26-3" name="__codelineno-26-3" href="#__codelineno-26-3"></a>  <span class="nf">plot</span><span class="p">(</span><span class="n">logCPM</span><span class="p">[,</span><span class="n">i</span><span class="p">],</span> <span class="n">q</span><span class="o">$</span><span class="n">E</span><span class="p">[,</span><span class="n">i</span><span class="p">],</span> <span class="n">xlab</span><span class="o">=</span><span class="s">&quot;LogCPM&quot;</span><span class="p">,</span>
<a id="__codelineno-26-4" name="__codelineno-26-4" href="#__codelineno-26-4"></a>       <span class="n">ylab</span><span class="o">=</span><span class="s">&quot;Voom (+quantile norm)&quot;</span><span class="p">,</span><span class="n">main</span><span class="o">=</span><span class="nf">colnames</span><span class="p">(</span><span class="n">logCPM</span><span class="p">)[</span><span class="n">i</span><span class="p">])</span>
<a id="__codelineno-26-5" name="__codelineno-26-5" href="#__codelineno-26-5"></a>  <span class="nf">abline</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">1</span><span class="p">)</span>
<a id="__codelineno-26-6" name="__codelineno-26-6" href="#__codelineno-26-6"></a><span class="p">}</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-21-1.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-27-1" name="__codelineno-27-1" href="#__codelineno-27-1"></a><span class="nf">boxplot</span><span class="p">(</span><span class="n">q</span><span class="o">$</span><span class="n">E</span> <span class="o">~</span> <span class="nf">col</span><span class="p">(</span><span class="n">q</span><span class="o">$</span><span class="n">E</span><span class="p">))</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-22-1.png" /><!-- --></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-28-1" name="__codelineno-28-1" href="#__codelineno-28-1"></a><span class="n">lineColour</span> <span class="o">&lt;-</span> <span class="nf">ifelse</span><span class="p">(</span><span class="n">conds</span><span class="o">==</span><span class="s">&quot;MT&quot;</span><span class="p">,</span> <span class="s">&quot;red&quot;</span><span class="p">,</span> <span class="s">&quot;blue&quot;</span><span class="p">)</span>
<a id="__codelineno-28-2" name="__codelineno-28-2" href="#__codelineno-28-2"></a><span class="n">lineColour</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] &quot;blue&quot; &quot;blue&quot; &quot;blue&quot; &quot;red&quot;  &quot;red&quot;  &quot;red&quot;
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-29-1" name="__codelineno-29-1" href="#__codelineno-29-1"></a><span class="nf">plot</span><span class="p">(</span><span class="nf">density</span><span class="p">(</span><span class="n">q</span><span class="o">$</span><span class="n">E</span><span class="p">[,</span><span class="m">1</span><span class="p">]),</span> <span class="n">ylim</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0.3</span><span class="p">),</span> <span class="n">col</span><span class="o">=</span><span class="n">lineColour</span><span class="p">[</span><span class="m">1</span><span class="p">])</span>
<a id="__codelineno-29-2" name="__codelineno-29-2" href="#__codelineno-29-2"></a><span class="nf">for</span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">2</span><span class="o">:</span><span class="nf">ncol</span><span class="p">(</span><span class="n">logCPM</span><span class="p">))</span> <span class="nf">lines</span><span class="p">(</span><span class="nf">density</span><span class="p">(</span><span class="n">q</span><span class="o">$</span><span class="n">E</span><span class="p">[,</span><span class="n">i</span><span class="p">]),</span> <span class="n">col</span><span class="o">=</span><span class="n">lineColour</span><span class="p">[</span><span class="n">i</span><span class="p">])</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-23-1.png" /><!-- --></p>
<p><strong>Note: we’re NOT going to use the quantile normalised data here, but I
wanted to show you how that method works.</strong></p>
<h3 id="statistical-testing-fitting-a-linear-model">Statistical testing: fitting a linear model<a class="headerlink" href="#statistical-testing-fitting-a-linear-model" title="Permanent link">&para;</a></h3>
<p>The next step is to fit a linear model to the transformed count data.</p>
<p>The <code>lmFit()</code> command does this <em>for each gene</em> (all at once).</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-30-1" name="__codelineno-30-1" href="#__codelineno-30-1"></a><span class="n">fit</span> <span class="o">&lt;-</span> <span class="nf">lmFit</span><span class="p">(</span><span class="n">v</span><span class="p">,</span> <span class="n">design</span><span class="p">)</span>
</code></pre></div>
<p>The <code>eBayes()</code> function then performs Emprical Bays shrinkage estimation
to adjust the variability estimates for each gene, and produce the
moderated t-statistics.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-31-1" name="__codelineno-31-1" href="#__codelineno-31-1"></a><span class="n">fit</span> <span class="o">&lt;-</span> <span class="nf">eBayes</span><span class="p">(</span><span class="n">fit</span><span class="p">)</span>
</code></pre></div>
<p>We then summarize this information using the <code>topTable()</code> function,
which list the genes in order of the strength of statistical support for
differential expression (i.e., genes with lowest p-values are at the top
of the list):</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-32-1" name="__codelineno-32-1" href="#__codelineno-32-1"></a><span class="n">tt</span> <span class="o">&lt;-</span> <span class="nf">topTable</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span> <span class="n">coef</span><span class="o">=</span><span class="nf">ncol</span><span class="p">(</span><span class="n">design</span><span class="p">),</span> <span class="n">n</span><span class="o">=</span><span class="nf">nrow</span><span class="p">(</span><span class="n">counts</span><span class="p">))</span>
<a id="__codelineno-32-2" name="__codelineno-32-2" href="#__codelineno-32-2"></a><span class="nf">head</span><span class="p">(</span><span class="n">tt</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##            logFC  AveExpr        t      P.Value    adj.P.Val        B
## YAL038W 2.313985 10.80214 319.5950 3.725452e-13 8.850433e-10 21.08951
## YOR161C 2.568389 10.80811 321.9628 3.574510e-13 8.850433e-10 21.08187
## YML128C 1.640664 11.40819 286.6167 6.857932e-13 9.775297e-10 20.84520
## YMR105C 2.772539  9.65092 331.8249 3.018547e-13 8.850433e-10 20.16815
## YHL021C 2.034496 10.17510 269.4034 9.702963e-13 1.152550e-09 20.07857
## YDR516C 2.085424 10.05426 260.8061 1.163655e-12 1.184767e-09 19.87217
</code></pre></div>
<h4 id="significant-genes">Significant genes<a class="headerlink" href="#significant-genes" title="Permanent link">&para;</a></h4>
<p>Adjusted p-values less than 0.05:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-33-1" name="__codelineno-33-1" href="#__codelineno-33-1"></a><span class="nf">sum</span><span class="p">(</span><span class="n">tt</span><span class="o">$</span><span class="n">adj.P.Val</span> <span class="o">&lt;</span> <span class="m">0.05</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 5140
</code></pre></div>
<p>Adjusted p-values less than 0.01:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-34-1" name="__codelineno-34-1" href="#__codelineno-34-1"></a><span class="nf">sum</span><span class="p">(</span><span class="n">tt</span><span class="o">$</span><span class="n">adj.P.Val</span> <span class="o">&lt;=</span> <span class="m">0.01</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 4566
</code></pre></div>
<p>By default, limma uses FDR adjustment (but lets check):</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-35-1" name="__codelineno-35-1" href="#__codelineno-35-1"></a><span class="nf">sum</span><span class="p">(</span><span class="nf">p.adjust</span><span class="p">(</span><span class="n">tt</span><span class="o">$</span><span class="n">P.Value</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s">&quot;fdr&quot;</span><span class="p">)</span> <span class="o">&lt;=</span> <span class="m">0.01</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 4566
</code></pre></div>
<p>Volcano plots are a popular method for summarising the <code>limma</code> output:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-36-1" name="__codelineno-36-1" href="#__codelineno-36-1"></a><span class="nf">volcanoplot</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span> <span class="n">coef</span><span class="o">=</span><span class="m">2</span><span class="p">)</span>
<a id="__codelineno-36-2" name="__codelineno-36-2" href="#__codelineno-36-2"></a><span class="nf">abline</span><span class="p">(</span><span class="n">h</span><span class="o">=-</span><span class="nf">log10</span><span class="p">(</span><span class="m">0.05</span><span class="p">))</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-32-1.png" /><!-- --></p>
<p>Significantly DE genes:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-37-1" name="__codelineno-37-1" href="#__codelineno-37-1"></a><span class="n">sigGenes</span> <span class="o">=</span> <span class="nf">which</span><span class="p">(</span><span class="n">tt</span><span class="o">$</span><span class="n">adj.P.Val</span> <span class="o">&lt;=</span> <span class="m">0.05</span><span class="p">)</span>
<a id="__codelineno-37-2" name="__codelineno-37-2" href="#__codelineno-37-2"></a><span class="nf">length</span><span class="p">(</span><span class="n">sigGenes</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 5140
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-38-1" name="__codelineno-38-1" href="#__codelineno-38-1"></a><span class="nf">volcanoplot</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span> <span class="n">coef</span><span class="o">=</span><span class="m">2</span><span class="p">)</span>
<a id="__codelineno-38-2" name="__codelineno-38-2" href="#__codelineno-38-2"></a><span class="nf">points</span><span class="p">(</span><span class="n">tt</span><span class="o">$</span><span class="n">logFC</span><span class="p">[</span><span class="n">sigGenes</span><span class="p">],</span> <span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">tt</span><span class="o">$</span><span class="n">P.Value</span><span class="p">[</span><span class="n">sigGenes</span><span class="p">]),</span> <span class="n">col</span><span class="o">=</span><span class="s">&#39;red&#39;</span><span class="p">,</span> <span class="n">pch</span><span class="o">=</span><span class="m">16</span><span class="p">,</span> <span class="n">cex</span><span class="o">=</span><span class="m">0.5</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-33-1.png" /><!-- --></p>
<p>Significantly DE genes above fold-change threshold:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-39-1" name="__codelineno-39-1" href="#__codelineno-39-1"></a><span class="n">sigGenes</span> <span class="o">=</span> <span class="nf">which</span><span class="p">(</span><span class="n">tt</span><span class="o">$</span><span class="n">adj.P.Val</span> <span class="o">&lt;=</span> <span class="m">0.05</span> <span class="o">&amp;</span> <span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">tt</span><span class="o">$</span><span class="n">logFC</span><span class="p">)</span> <span class="o">&gt;</span> <span class="nf">log2</span><span class="p">(</span><span class="m">2</span><span class="p">)))</span>
<a id="__codelineno-39-2" name="__codelineno-39-2" href="#__codelineno-39-2"></a><span class="nf">length</span><span class="p">(</span><span class="n">sigGenes</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 1891
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-40-1" name="__codelineno-40-1" href="#__codelineno-40-1"></a><span class="nf">volcanoplot</span><span class="p">(</span><span class="n">fit</span><span class="p">,</span> <span class="n">coef</span><span class="o">=</span><span class="m">2</span><span class="p">)</span>
<a id="__codelineno-40-2" name="__codelineno-40-2" href="#__codelineno-40-2"></a><span class="nf">points</span><span class="p">(</span><span class="n">tt</span><span class="o">$</span><span class="n">logFC</span><span class="p">[</span><span class="n">sigGenes</span><span class="p">],</span> <span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">tt</span><span class="o">$</span><span class="n">P.Value</span><span class="p">[</span><span class="n">sigGenes</span><span class="p">]),</span> <span class="n">col</span><span class="o">=</span><span class="s">&#39;red&#39;</span><span class="p">,</span> <span class="n">pch</span><span class="o">=</span><span class="m">16</span><span class="p">,</span> <span class="n">cex</span><span class="o">=</span><span class="m">0.5</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-34-1.png" /><!-- --></p>
<p>Get the rows of top table with significant adjusted p-values - we’ll
save these for later to compare with the other methods.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-41-1" name="__codelineno-41-1" href="#__codelineno-41-1"></a><span class="n">limmaPadj</span> <span class="o">&lt;-</span> <span class="n">tt</span><span class="p">[</span><span class="n">tt</span><span class="o">$</span><span class="n">adj.P.Val</span> <span class="o">&lt;=</span> <span class="m">0.01</span><span class="p">,</span> <span class="p">]</span>
</code></pre></div>
<hr />
<h3 id="deseq2">DESeq2<a class="headerlink" href="#deseq2" title="Permanent link">&para;</a></h3>
<ul>
<li>The DESeq2 package uses the <em>Negative Binomial</em> distribution to
    model the count data from each sample.</li>
<li>A statistical test based on the Negative Binomial distribution (via
    a generalized linear model, GLM) can be used to assess differential
    expression for each gene.</li>
<li>Use of the Negative Binomial distribution attempts to accurately
    capture the variation that is observed for count data.</li>
</ul>
<p>More information about DESeq2:
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">article
by Love et al, 2014</a></p>
<p>Set up “DESeq object” for analysis:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-42-1" name="__codelineno-42-1" href="#__codelineno-42-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">DESeq2</span><span class="p">)</span>
<a id="__codelineno-42-2" name="__codelineno-42-2" href="#__codelineno-42-2"></a><span class="n">dds</span> <span class="o">&lt;-</span> <span class="nf">DESeqDataSetFromMatrix</span><span class="p">(</span><span class="n">countData</span> <span class="o">=</span> <span class="nf">as.matrix</span><span class="p">(</span><span class="n">counts</span><span class="p">),</span> 
<a id="__codelineno-42-3" name="__codelineno-42-3" href="#__codelineno-42-3"></a>                              <span class="n">colData</span> <span class="o">=</span> <span class="nf">data.frame</span><span class="p">(</span><span class="n">conds</span><span class="o">=</span><span class="nf">factor</span><span class="p">(</span><span class="n">conds</span><span class="p">)),</span> 
<a id="__codelineno-42-4" name="__codelineno-42-4" href="#__codelineno-42-4"></a>                              <span class="n">design</span> <span class="o">=</span> <span class="nf">formula</span><span class="p">(</span><span class="o">~</span><span class="n">conds</span><span class="p">))</span>
</code></pre></div>
<p>Can access the count data in teh <code>dds</code> object via the <code>counts()</code>
function:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-43-1" name="__codelineno-43-1" href="#__codelineno-43-1"></a><span class="nf">counts</span><span class="p">(</span><span class="n">dds</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">head</span><span class="p">()</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##           WT1 WT2 WT3 MT1 MT2 MT3
## YDL248W    52  46  36  65  70  78
## YDL247W-A   0   0   0   0   1   0
## YDL247W     2   4   2   6   8   5
## YDL246C     0   0   1   1   2   0
## YDL245C     0   3   0   5   7   4
## YDL244W     6   6   5  20  30  19
</code></pre></div>
<p>Fit DESeq model to identify DE transcripts</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-44-1" name="__codelineno-44-1" href="#__codelineno-44-1"></a><span class="n">dds</span> <span class="o">&lt;-</span> <span class="nf">DESeq</span><span class="p">(</span><span class="n">dds</span><span class="p">)</span>
</code></pre></div>
<p>Take a look at the results table</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-45-1" name="__codelineno-45-1" href="#__codelineno-45-1"></a><span class="n">res</span> <span class="o">&lt;-</span> <span class="n">DESeq2</span><span class="o">::</span><span class="nf">results</span><span class="p">(</span><span class="n">dds</span><span class="p">)</span>
<a id="__codelineno-45-2" name="__codelineno-45-2" href="#__codelineno-45-2"></a><span class="n">knitr</span><span class="o">::</span><span class="nf">kable</span><span class="p">(</span><span class="n">res</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">6</span><span class="p">,])</span>
</code></pre></div>
<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">lfcSE</th>
<th align="right">stat</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">YDL248W</td>
<td align="right">56.2230316</td>
<td align="right">-0.2339470</td>
<td align="right">0.2278417</td>
<td align="right">-1.0267961</td>
<td align="right">0.3045165</td>
<td align="right">0.3663639</td>
</tr>
<tr>
<td align="left">YDL247W-A</td>
<td align="right">0.1397714</td>
<td align="right">-0.5255769</td>
<td align="right">4.0804729</td>
<td align="right">-0.1288030</td>
<td align="right">0.8975136</td>
<td align="right">0.9173926</td>
</tr>
<tr>
<td align="left">YDL247W</td>
<td align="right">4.2428211</td>
<td align="right">-0.8100552</td>
<td align="right">0.8332543</td>
<td align="right">-0.9721584</td>
<td align="right">0.3309718</td>
<td align="right">0.3948536</td>
</tr>
<tr>
<td align="left">YDL246C</td>
<td align="right">0.6182409</td>
<td align="right">-1.0326364</td>
<td align="right">2.1560899</td>
<td align="right">-0.4789394</td>
<td align="right">0.6319817</td>
<td align="right">0.6915148</td>
</tr>
<tr>
<td align="left">YDL245C</td>
<td align="right">2.8486580</td>
<td align="right">-1.9781751</td>
<td align="right">1.1758845</td>
<td align="right">-1.6822869</td>
<td align="right">0.0925132</td>
<td align="right">0.1230028</td>
</tr>
<tr>
<td align="left">YDL244W</td>
<td align="right">13.0883354</td>
<td align="right">-1.5823354</td>
<td align="right">0.5128788</td>
<td align="right">-3.0852037</td>
<td align="right">0.0020341</td>
<td align="right">0.0032860</td>
</tr>
</tbody>
</table>
<p>Summary of differential gene expression</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-46-1" name="__codelineno-46-1" href="#__codelineno-46-1"></a><span class="nf">summary</span><span class="p">(</span><span class="n">res</span><span class="p">)</span> 
</code></pre></div>
<div class="highlight"><pre><span></span><code>## 
## out of 6830 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 2520, 37%
## LFC &lt; 0 (down)     : 2521, 37%
## outliers [1]       : 0, 0%
## low counts [2]     : 0, 0%
## (mean count &lt; 0)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results
</code></pre></div>
<p>Note: p-value adjustment</p>
<ul>
<li>The “padj” column of the DESeq2 results (<code>res</code>) contains adjusted
    p-values (FDR).</li>
<li>Can use the <code>p.adjust</code> function to manually adjust the DESeq2
    p-values if needed (e.g., to use Holm correction)</li>
</ul>
<div class="highlight"><pre><span></span><code><a id="__codelineno-47-1" name="__codelineno-47-1" href="#__codelineno-47-1"></a><span class="c1"># Remove rows with NAs</span>
<a id="__codelineno-47-2" name="__codelineno-47-2" href="#__codelineno-47-2"></a><span class="n">res</span> <span class="o">=</span> <span class="nf">na.omit</span><span class="p">(</span><span class="n">res</span><span class="p">)</span>
<a id="__codelineno-47-3" name="__codelineno-47-3" href="#__codelineno-47-3"></a>
<a id="__codelineno-47-4" name="__codelineno-47-4" href="#__codelineno-47-4"></a><span class="c1"># Get the rows of &quot;res&quot; with significant adjusted p-values</span>
<a id="__codelineno-47-5" name="__codelineno-47-5" href="#__codelineno-47-5"></a><span class="n">resPadj</span><span class="o">&lt;-</span><span class="n">res</span><span class="p">[</span><span class="n">res</span><span class="o">$</span><span class="n">padj</span> <span class="o">&lt;=</span> <span class="m">0.05</span> <span class="p">,</span> <span class="p">]</span>
<a id="__codelineno-47-6" name="__codelineno-47-6" href="#__codelineno-47-6"></a>
<a id="__codelineno-47-7" name="__codelineno-47-7" href="#__codelineno-47-7"></a><span class="c1"># Get dimensions</span>
<a id="__codelineno-47-8" name="__codelineno-47-8" href="#__codelineno-47-8"></a><span class="nf">dim</span><span class="p">(</span><span class="n">resPadj</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 4811    6
</code></pre></div>
<p>Number of adjusted p-values less than 0.05</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-48-1" name="__codelineno-48-1" href="#__codelineno-48-1"></a><span class="nf">sum</span><span class="p">(</span><span class="n">res</span><span class="o">$</span><span class="n">padj</span> <span class="o">&lt;=</span> <span class="m">0.05</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 4811
</code></pre></div>
<p>Check that this is the same using p.adjust with FDR correction</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-49-1" name="__codelineno-49-1" href="#__codelineno-49-1"></a><span class="nf">sum</span><span class="p">(</span><span class="nf">p.adjust</span><span class="p">(</span><span class="n">res</span><span class="o">$</span><span class="n">pvalue</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s">&quot;fdr&quot;</span><span class="p">)</span> <span class="o">&lt;=</span> <span class="m">0.05</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 4811
</code></pre></div>
<p>Number of Holm adjusted p-values less than 0.05</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-50-1" name="__codelineno-50-1" href="#__codelineno-50-1"></a><span class="nf">sum</span><span class="p">(</span><span class="nf">p.adjust</span><span class="p">(</span><span class="n">res</span><span class="o">$</span><span class="n">pvalue</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s">&quot;holm&quot;</span><span class="p">)</span> <span class="o">&lt;=</span> <span class="m">0.01</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 3429
</code></pre></div>
<p>Sort summary list by p-value and save the table as a CSV file that can
be read in Excel (or any other spreadhseet program).</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-51-1" name="__codelineno-51-1" href="#__codelineno-51-1"></a><span class="n">res</span> <span class="o">&lt;-</span> <span class="n">res</span><span class="p">[</span><span class="nf">order</span><span class="p">(</span><span class="n">res</span><span class="o">$</span><span class="n">padj</span><span class="p">),]</span>
<a id="__codelineno-51-2" name="__codelineno-51-2" href="#__codelineno-51-2"></a>
<a id="__codelineno-51-3" name="__codelineno-51-3" href="#__codelineno-51-3"></a><span class="nf">write.csv</span><span class="p">(</span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">res</span><span class="p">),</span><span class="n">file</span><span class="o">=</span><span class="s">&#39;deseq2.csv&#39;</span><span class="p">)</span>
</code></pre></div>
<hr />
<h3 id="edger">edgeR<a class="headerlink" href="#edger" title="Permanent link">&para;</a></h3>
<ul>
<li>The edgeR package also uses the negative binomial distribution to
    model the RNA-seq count data.</li>
<li>Takes an empirical Bayes approach to the statistical analysis, in a
    similar way to how the <code>limma</code> package handles RNA-seq data.</li>
</ul>
<h4 id="identify-degs-with-edgers-exact-method">Identify DEGs with edgeR’s “Exact” Method<a class="headerlink" href="#identify-degs-with-edgers-exact-method" title="Permanent link">&para;</a></h4>
<p>Load <code>edgeR</code> package</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-52-1" name="__codelineno-52-1" href="#__codelineno-52-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">edgeR</span><span class="p">)</span>
</code></pre></div>
<p>Construct <code>DGEList</code> object</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-53-1" name="__codelineno-53-1" href="#__codelineno-53-1"></a><span class="n">y</span> <span class="o">&lt;-</span> <span class="nf">DGEList</span><span class="p">(</span><span class="n">counts</span><span class="o">=</span><span class="n">counts</span><span class="p">,</span> <span class="n">group</span><span class="o">=</span><span class="n">conds</span><span class="p">)</span>
</code></pre></div>
<p>Calculate library size (counts per sample)</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-54-1" name="__codelineno-54-1" href="#__codelineno-54-1"></a><span class="n">y</span> <span class="o">&lt;-</span> <span class="nf">calcNormFactors</span><span class="p">(</span><span class="n">y</span><span class="p">)</span>
</code></pre></div>
<p>Estimate common dispersion (overall variability)</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-55-1" name="__codelineno-55-1" href="#__codelineno-55-1"></a><span class="n">y</span> <span class="o">&lt;-</span> <span class="nf">estimateCommonDisp</span><span class="p">(</span><span class="n">y</span><span class="p">)</span>
</code></pre></div>
<p>Estimate tagwise dispersion (per gene variability)</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-56-1" name="__codelineno-56-1" href="#__codelineno-56-1"></a><span class="n">y</span> <span class="o">&lt;-</span> <span class="nf">estimateTagwiseDisp</span><span class="p">(</span><span class="n">y</span><span class="p">)</span>
</code></pre></div>
<h4 id="edger-analysis-and-output">edgeR analysis and output<a class="headerlink" href="#edger-analysis-and-output" title="Permanent link">&para;</a></h4>
<p>Compute exact test for the negative binomial distribution.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-57-1" name="__codelineno-57-1" href="#__codelineno-57-1"></a><span class="n">et</span> <span class="o">&lt;-</span> <span class="nf">exactTest</span><span class="p">(</span><span class="n">y</span><span class="p">)</span> 
<a id="__codelineno-57-2" name="__codelineno-57-2" href="#__codelineno-57-2"></a>
<a id="__codelineno-57-3" name="__codelineno-57-3" href="#__codelineno-57-3"></a><span class="n">knitr</span><span class="o">::</span><span class="nf">kable</span><span class="p">(</span><span class="nf">topTags</span><span class="p">(</span><span class="n">et</span><span class="p">,</span> <span class="n">n</span><span class="o">=</span><span class="m">4</span><span class="p">)</span><span class="o">$</span><span class="n">table</span><span class="p">)</span>
</code></pre></div>
<table>
<thead>
<tr>
<th align="left"></th>
<th align="right">logFC</th>
<th align="right">logCPM</th>
<th align="right">PValue</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">YCR077C</td>
<td align="right">13.098599</td>
<td align="right">6.629306</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr>
<td align="left">snR71</td>
<td align="right">-9.870811</td>
<td align="right">8.294459</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr>
<td align="left">snR59</td>
<td align="right">-8.752555</td>
<td align="right">10.105453</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr>
<td align="left">snR53</td>
<td align="right">-8.503101</td>
<td align="right">7.687908</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p><em>Adjusted p-values</em></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-58-1" name="__codelineno-58-1" href="#__codelineno-58-1"></a><span class="n">edge</span> <span class="o">&lt;-</span> <span class="nf">as.data.frame</span><span class="p">(</span><span class="nf">topTags</span><span class="p">(</span><span class="n">et</span><span class="p">,</span> <span class="n">n</span><span class="o">=</span><span class="nf">nrow</span><span class="p">(</span><span class="n">counts</span><span class="p">)))</span> 
<a id="__codelineno-58-2" name="__codelineno-58-2" href="#__codelineno-58-2"></a><span class="nf">sum</span><span class="p">(</span><span class="n">edge</span><span class="o">$</span><span class="n">FDR</span> <span class="o">&lt;=</span> <span class="m">0.05</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 5242
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-59-1" name="__codelineno-59-1" href="#__codelineno-59-1"></a><span class="nf">sum</span><span class="p">(</span><span class="nf">p.adjust</span><span class="p">(</span><span class="n">edge</span><span class="o">$</span><span class="n">PValue</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s">&quot;fdr&quot;</span><span class="p">)</span> <span class="o">&lt;=</span> <span class="m">0.05</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 5242
</code></pre></div>
<p>Get the rows of “edge” with significant adjusted p-values</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-60-1" name="__codelineno-60-1" href="#__codelineno-60-1"></a><span class="n">edgePadj</span> <span class="o">&lt;-</span> <span class="n">edge</span><span class="p">[</span><span class="n">edge</span><span class="o">$</span><span class="n">FDR</span> <span class="o">&lt;=</span> <span class="m">0.05</span><span class="p">,</span> <span class="p">]</span>
</code></pre></div>
<h3 id="deseq2-vs-edger">DESeq2 vs edgeR<a class="headerlink" href="#deseq2-vs-edger" title="Permanent link">&para;</a></h3>
<p>Generate Venn diagram to compare DESeq2 and edgeR results.</p>
<p>There is fairly good overlap:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-61-1" name="__codelineno-61-1" href="#__codelineno-61-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">gplots</span><span class="p">)</span>
<a id="__codelineno-61-2" name="__codelineno-61-2" href="#__codelineno-61-2"></a><span class="n">setlist</span> <span class="o">&lt;-</span> <span class="nf">list</span><span class="p">(</span><span class="n">edgeRexact</span><span class="o">=</span><span class="nf">rownames</span><span class="p">(</span><span class="n">edgePadj</span><span class="p">),</span> <span class="n">DESeq2</span><span class="o">=</span><span class="nf">rownames</span><span class="p">(</span><span class="n">resPadj</span><span class="p">))</span>
<a id="__codelineno-61-3" name="__codelineno-61-3" href="#__codelineno-61-3"></a><span class="nf">venn</span><span class="p">(</span><span class="n">setlist</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-54-1.png" /><!-- --></p>
<hr />
<h3 id="limma-vs-edger-vs-deseq2">Limma vs edgeR vs DESeq2<a class="headerlink" href="#limma-vs-edger-vs-deseq2" title="Permanent link">&para;</a></h3>
<p>Again, fairly good overlap across the three methods we’ve looked at:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-62-1" name="__codelineno-62-1" href="#__codelineno-62-1"></a><span class="n">setlist</span> <span class="o">&lt;-</span> <span class="nf">list</span><span class="p">(</span><span class="n">edgeRexact</span><span class="o">=</span><span class="nf">rownames</span><span class="p">(</span><span class="n">edgePadj</span><span class="p">),</span> 
<a id="__codelineno-62-2" name="__codelineno-62-2" href="#__codelineno-62-2"></a>                <span class="n">DESeq2</span><span class="o">=</span><span class="nf">rownames</span><span class="p">(</span><span class="n">resPadj</span><span class="p">),</span>
<a id="__codelineno-62-3" name="__codelineno-62-3" href="#__codelineno-62-3"></a>                <span class="n">LimmaVoom</span><span class="o">=</span><span class="nf">rownames</span><span class="p">(</span><span class="n">limmaPadj</span><span class="p">))</span>
<a id="__codelineno-62-4" name="__codelineno-62-4" href="#__codelineno-62-4"></a><span class="nf">venn</span><span class="p">(</span><span class="n">setlist</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-diffexp_files/figure-gfm/unnamed-chunk-56-1.png" /><!-- --></p>
<h3 id="summary">Summary<a class="headerlink" href="#summary" title="Permanent link">&para;</a></h3>
<ul>
<li>Once we’ve generated count data, there are a number of ways to
    perform a differential expression analysis.<ul>
<li>DESeq2 and edgeR model the count data, and assume a Negative
    Binomial distribution</li>
<li>Limma transforms (and logs) the data and assumes normality</li>
</ul>
</li>
<li>Here we’ve seen that these three approaches give quite similar
    results.</li>
<li>The next step in a “standard” RNA-seq workflow is to perform
    “pathway analysis”.</li>
</ul>
<h4 id="save-limma-toptable-results-for-next-session">Save limma topTable results for next session…<a class="headerlink" href="#save-limma-toptable-results-for-next-session" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-63-1" name="__codelineno-63-1" href="#__codelineno-63-1"></a><span class="nf">save</span><span class="p">(</span><span class="n">list</span><span class="o">=</span><span class="s">&#39;tt&#39;</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s">&#39;topTable.RData&#39;</span><span class="p">)</span>
</code></pre></div>
<hr />
<p align="center"><b><a class="btn" href="https://genomicsaotearoa.github.io/RNA-seq-workshop/" style="background: var(--bs-dark);font-weight:bold">Back to homepage</a></b></p>

              
            </article>
            
          </div>
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
        
        <a href="../4_mappingANDcount/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Mapping and Counting" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Mapping and Counting
            </div>
          </div>
        </a>
      
      
        
        <a href="../6_over-representation_analysis/" class="md-footer__link md-footer__link--next" aria-label="Next: Over-representation analysis (Gene Ontology)" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Over-representation analysis (Gene Ontology)
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top"], "search": "../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.6c7ad80a.min.js"></script>
      
        <script src="../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>