---
layout: homepage
---

This RNA-Seq workshop aims to get you started with your own RNA-seq analysis and assumes you are already familiar with the basics of bash and R.

## Content

1. [Background](0.Background/README.md)
2. [Quality Assessment](./1.Quality_Assessment/README.md)
3. [Trimming and Filtering](./2.Trimming_And_Filtering/README.md)
4. [Mapping and Count](./3.Mapping_And_Count/README.md)
5. [Differential Expression Analysis](./4.Differential_Expression/rnaseq-diffexp.md)
6. [Over-representation Analysis](./5.Overrepresentation_Analysis/rnaseq-pathway.md)
7. [Supplementary 1 - A Guide to RNA-Seq](./6.Supplementary/supp_1.md)
8. [Supplementary 2 - Lecture notes : DE analysis](./6.Supplementary/supp_2.md)
9. [Supplementary 3 - Lecture notes : Annotation Data & Gene Set Analysis](./6.Supplementary/supp_3.md)

## Attribution notice
  
- The material used to prepare for the workshop was extracted from Assoc. Prof. Michael Black lectures for Stat435 course.

- Some code used in this workshop has been extracted from a collection of presentations, workshops and lectures by Prof. Thomas Girke, Institute for Integrative Genome Biology, UC Riverside: <a href="http://girke.bioinformatics.ucr.edu/">Link_1</a> and <a href="http://faculty.ucr.edu/~tgirke/HTML_Presentations/Manuals/MCBIOS2015/Rrnaseq/Rrnaseq.pdf">Link_2</a>
