
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa & NeSI">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/RNA-seq-workshop/6_over-representation_analysis/">
      
      <link rel="icon" href="../Images/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.3.0, mkdocs-material-8.3.9">
    
    
      
        <title>Over-representation analysis (Gene Ontology) - RNA-Seq Data Analysis</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.1d29e8d0.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.cbb835fc.min.css">
        
      
      
    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CFira+Code:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Fira Code"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="green" data-md-color-accent="">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#over-representation-analysis-gene-ontology" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="RNA-Seq Data Analysis" class="md-header__button md-logo" aria-label="RNA-Seq Data Analysis" data-md-component="logo">
      
  <img src="../Images/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            RNA-Seq Data Analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Over-representation analysis (Gene Ontology)
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="" data-md-color-primary="green" data-md-color-accent=""  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_3" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="green" data-md-color-accent=""  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_2">
          
            <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="green" data-md-color-accent=""  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_3">
          
            <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_2" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
            </label>
          
        
      </form>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/RNA-seq-workshop.git" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/RNA-seq-workshop
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="RNA-Seq Data Analysis" class="md-nav__button md-logo" aria-label="RNA-Seq Data Analysis" data-md-component="logo">
      
  <img src="../Images/nesi_ga.png" alt="logo">

    </a>
    RNA-Seq Data Analysis
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/RNA-seq-workshop.git" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/RNA-seq-workshop
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../1_background/" class="md-nav__link">
        RNA-seq Workflow
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../2_qualityassessment/" class="md-nav__link">
        Quality control of the sequencing data.
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../3_trimmingfiltering/" class="md-nav__link">
        Cleaning Reads
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../4_mappingANDcount/" class="md-nav__link">
        Mapping and Counting
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../5_rnaseq-diffexp/" class="md-nav__link">
        Differential Expression Analysis
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Over-representation analysis (Gene Ontology)
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Over-representation analysis (Gene Ontology)
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#goseq" class="md-nav__link">
    GOseq
  </a>
  
    <nav class="md-nav" aria-label="GOseq">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#results-from-the-young-et-al-2010-publication" class="md-nav__link">
    Results from the Young et al (2010) publication:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#goseq-analysis" class="md-nav__link">
    GOseq analysis
  </a>
  
    <nav class="md-nav" aria-label="GOseq analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#calculate-gene-weights" class="md-nav__link">
    Calculate gene weights
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inspect-output" class="md-nav__link">
    Inspect output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-lengths-and-weights" class="md-nav__link">
    Gene lengths and weights
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-length-vs-average-expression" class="md-nav__link">
    Gene length vs average expression
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#length-correction-in-goseq" class="md-nav__link">
    Length correction in GOSeq
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#output-wallenius-method" class="md-nav__link">
    Output: Wallenius method
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#p-value-adjustment" class="md-nav__link">
    P-value adjustment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering-by-gene-set-length" class="md-nav__link">
    Filtering by gene set length
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#go-terms" class="md-nav__link">
    GO terms
  </a>
  
    <nav class="md-nav" aria-label="GO terms">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-goseq-without-gene-length-correction" class="md-nav__link">
    Run GOSeq without gene length correction
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#p-value-adjustment_1" class="md-nav__link">
    P-value adjustment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#compare-with-and-without-adjustment" class="md-nav__link">
    Compare with and without adjustment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-lengths-and-go-term-membership" class="md-nav__link">
    Gene lengths and GO term membership
  </a>
  
    <nav class="md-nav" aria-label="Gene lengths and GO term membership">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#getting-fancy" class="md-nav__link">
    Getting fancy…
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#significant-hypergeometric-vs-wallenius" class="md-nav__link">
    Significant: Hypergeometric vs Wallenius
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-significant-go-terms" class="md-nav__link">
    All significant GO terms
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-length-versus-p-value" class="md-nav__link">
    Gene length versus P-value
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#summary" class="md-nav__link">
    Summary
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../supp_1/" class="md-nav__link">
        Supplementary 1 - A Guide to RNA-Seq
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../supp_2/" class="md-nav__link">
        Supplementary 2 -Lecture notes : Introduction To Differential Expression Analysis
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../supp_3/" class="md-nav__link">
        Supplementary 3 - Lecture notes : Annotation Data & Gene Set Analysis
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#goseq" class="md-nav__link">
    GOseq
  </a>
  
    <nav class="md-nav" aria-label="GOseq">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#results-from-the-young-et-al-2010-publication" class="md-nav__link">
    Results from the Young et al (2010) publication:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#goseq-analysis" class="md-nav__link">
    GOseq analysis
  </a>
  
    <nav class="md-nav" aria-label="GOseq analysis">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#calculate-gene-weights" class="md-nav__link">
    Calculate gene weights
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inspect-output" class="md-nav__link">
    Inspect output
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-lengths-and-weights" class="md-nav__link">
    Gene lengths and weights
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-length-vs-average-expression" class="md-nav__link">
    Gene length vs average expression
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#length-correction-in-goseq" class="md-nav__link">
    Length correction in GOSeq
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#output-wallenius-method" class="md-nav__link">
    Output: Wallenius method
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#p-value-adjustment" class="md-nav__link">
    P-value adjustment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#filtering-by-gene-set-length" class="md-nav__link">
    Filtering by gene set length
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#go-terms" class="md-nav__link">
    GO terms
  </a>
  
    <nav class="md-nav" aria-label="GO terms">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#run-goseq-without-gene-length-correction" class="md-nav__link">
    Run GOSeq without gene length correction
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#p-value-adjustment_1" class="md-nav__link">
    P-value adjustment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#compare-with-and-without-adjustment" class="md-nav__link">
    Compare with and without adjustment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-lengths-and-go-term-membership" class="md-nav__link">
    Gene lengths and GO term membership
  </a>
  
    <nav class="md-nav" aria-label="Gene lengths and GO term membership">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#getting-fancy" class="md-nav__link">
    Getting fancy…
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#significant-hypergeometric-vs-wallenius" class="md-nav__link">
    Significant: Hypergeometric vs Wallenius
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-significant-go-terms" class="md-nav__link">
    All significant GO terms
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gene-length-versus-p-value" class="md-nav__link">
    Gene length versus P-value
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#summary" class="md-nav__link">
    Summary
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
  <a href="https://github.com/GenomicsAotearoa/RNA-seq-workshop.git/edit/master/docs/6_over-representation_analysis.md" title="Edit this page" class="md-content__button md-icon">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
  </a>



<h1 id="over-representation-analysis-gene-ontology">Over-representation analysis (Gene Ontology)<a class="headerlink" href="#over-representation-analysis-gene-ontology" title="Permanent link">&para;</a></h1>
<div class="admonition info">
<p class="admonition-title">Objectives</p>
</div>
<p><center>
<img alt="image" src="../theme_images/over-rep_theme.png" width="250" />
</center></p>
<ul>
<li>Slides: <a href="https://github.com/GenomicsAotearoa/RNA-seq-workshop/blob/master/5.Overrepresentation_Analysis/lecture_annotation_pathways.pdf">lecture_annotation_pathways.pdf</a></li>
</ul>
<p>Over-representation Analysis
(<a href="https://academic.oup.com/bioinformatics/article/20/18/3710/202612">Boyle
et al. 2004</a>) is a widely used approach to determine whether known
biological functions or processes are over-represented (= enriched) in
an experimentally-derived gene list, e.g. a list of differentially
expressed genes (DEGs).</p>
<ul>
<li>Can perform over-representation analysis online (e.g., Enrichr,
    GeneSetDB, PantherDB), and also in R.</li>
<li>The basic principles are to:<ul>
<li>identify a collection of differentially expressed genes</li>
<li>test to see if genes that are members of specific gene sets
    (e.g., Reactome pathways, Gene Ontology categories) are
    differentially expressed more often than would be expected by
    chance.</li>
</ul>
</li>
</ul>
<div class="admonition abstract">
<p class="admonition-title">Some caveats for RNA-seq data</p>
<ul>
<li>The gene-set analysis methods are applicable to transcriptomic data
    from both microarrays and RNA-seq.</li>
<li>One caveat, however, is that the results need to take <em>gene length</em>
    into account.</li>
<li>RNA-seq tends to produce higher expression levels (i.e., greater
    counts) for longer genes: a longer transcript implies more aligned
    fragments, and thus higher counts. This also gives these genes a
    great chance of being statistically differentially expressed.</li>
<li>Some gene sets (pathways, GO terms) tend to involve families of long
    genes: if long genes have a great chance of being detected as
    differentially expressed, then <em>gene sets</em> consisting of long genes
    will have a great chance of appeared to be enriched in the analysis.</li>
</ul>
</div>
<h3 id="goseq">GOseq<a class="headerlink" href="#goseq" title="Permanent link">&para;</a></h3>
<ul>
<li>The GOseq methodology
    (<a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-2-r14">Young
    et al., 2010</a>) overcomes this issue by allowing the
    over-representation analysis to be adjusted for gene length.</li>
<li>modification to hypergeometric sampling probability</li>
<li>exact method (resampling) and an approximation-based method</li>
<li>More recent publications have also applied this gene-length
    correction to GSEA-based methods.</li>
<li>Still not widely understood to be an issue when performing RNA-seq
    pathway analysis, but REALLY important to take into account.</li>
</ul>
<h4 id="results-from-the-young-et-al-2010-publication">Results from the Young et al (2010) publication:<a class="headerlink" href="#results-from-the-young-et-al-2010-publication" title="Permanent link">&para;</a></h4>
<p><img alt="image" src="../PNG/goseq-1.png" />
<center><small>Proportion DE by gene length and reads</small></center></p>
<p><img alt="image" src="../PNG/goseq-2.png" />
<center><small>Gene set ranks by standard analysis</small></center></p>
<p><img alt="image" src="../PNG/goseq-3.png" />
<center><small>Gene set ranks by GOseq</small></center></p>
<h3 id="goseq-analysis">GOseq analysis<a class="headerlink" href="#goseq-analysis" title="Permanent link">&para;</a></h3>
<p>Need to figure out if our organism is supported… (code is “sacCer”)</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">dplyr</span><span class="p">)</span>
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a><span class="nf">library</span><span class="p">(</span><span class="n">goseq</span><span class="p">)</span>
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a><span class="nf">supportedOrganisms</span><span class="p">()</span> <span class="o">%&gt;%</span> <span class="nf">head</span><span class="p">()</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##          Genome         Id  Id Description Lengths in geneLeneDataBase
## 10      anoCar1    ensGene Ensembl gene ID                        TRUE
## 11      anoGam1    ensGene Ensembl gene ID                        TRUE
## 132     anoGam3                                                  FALSE
## 12      apiMel2    ensGene Ensembl gene ID                        TRUE
## 137 Arabidopsis                                                  FALSE
## 56      bosTau2 geneSymbol     Gene Symbol                        TRUE
##     GO Annotation Available
## 10                    FALSE
## 11                     TRUE
## 132                    TRUE
## 12                    FALSE
## 137                    TRUE
## 56                     TRUE
</code></pre></div>
<p>Easier to find if we use <code>View()</code> <strong>(NB - this only works in RStudio.
Can’t use in Jupyter on NeSI).</strong></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a><span class="nf">supportedOrganisms</span><span class="p">()</span> <span class="o">%&gt;%</span> <span class="nf">View</span><span class="p">()</span>
</code></pre></div>
<p><em>Define differentially expressed genes</em></p>
<ul>
<li>Create a vector of 0’s and 1’s to denote whether or not genes are
    differentially expressed (limma analysis: topTable).</li>
<li>Add gene names to the vector so that GOSeq knows which gene each
    data point relates to.</li>
</ul>
<p>Load our topTable results from last session:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="nf">load</span><span class="p">(</span><span class="s">&#39;topTable.RData&#39;</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="c1"># Note: If you want to use tt from DESeq, replace $adj.P.Val with $padj below</span>
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>
<a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a><span class="n">genes</span> <span class="o">&lt;-</span> <span class="nf">ifelse</span><span class="p">(</span><span class="n">tt</span><span class="o">$</span><span class="n">adj.P.Val</span> <span class="o">&lt;</span> <span class="m">0.05</span><span class="p">,</span> <span class="m">1</span><span class="p">,</span> <span class="m">0</span><span class="p">)</span> 
<a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="nf">names</span><span class="p">(</span><span class="n">genes</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="nf">rownames</span><span class="p">(</span><span class="n">tt</span><span class="p">)</span>
<a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a><span class="nf">head</span><span class="p">(</span><span class="n">genes</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## YAL038W YOR161C YML128C YMR105C YHL021C YDR516C 
##       1       1       1       1       1       1
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="nf">table</span><span class="p">(</span><span class="n">genes</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## genes
##    0    1 
## 1987 5140
</code></pre></div>
<h4 id="calculate-gene-weights">Calculate gene weights<a class="headerlink" href="#calculate-gene-weights" title="Permanent link">&para;</a></h4>
<ul>
<li>Put genes into length-based “bins”, and plot length vs proportion
    differentially expressed</li>
<li>Likely restricts to only those genes with GO annotation</li>
</ul>
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="n">pwf</span><span class="o">=</span><span class="nf">nullp</span><span class="p">(</span><span class="n">genes</span><span class="p">,</span> <span class="s">&quot;sacCer1&quot;</span><span class="p">,</span> <span class="s">&quot;ensGene&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-pathway_files/figure-gfm/unnamed-chunk-5-1.png" /><!-- --></p>
<h4 id="inspect-output">Inspect output<a class="headerlink" href="#inspect-output" title="Permanent link">&para;</a></h4>
<p>Report length (bias) and weight data per gene.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="nf">head</span><span class="p">(</span><span class="n">pwf</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##         DEgenes bias.data       pwf
## YAL038W       1      1504 0.8205505
## YOR161C       1      1621 0.8205505
## YML128C       1      1543 0.8205505
## YMR105C       1      1711 0.8205505
## YHL021C       1      1399 0.8205505
## YDR516C       1      1504 0.8205505
</code></pre></div>
<h4 id="gene-lengths-and-weights">Gene lengths and weights<a class="headerlink" href="#gene-lengths-and-weights" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="nf">par</span><span class="p">(</span><span class="n">mfrow</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">))</span>
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a><span class="nf">hist</span><span class="p">(</span><span class="n">pwf</span><span class="o">$</span><span class="n">bias.data</span><span class="p">,</span><span class="m">30</span><span class="p">)</span>
<a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a><span class="nf">hist</span><span class="p">(</span><span class="n">pwf</span><span class="o">$</span><span class="n">pwf</span><span class="p">,</span><span class="m">30</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-pathway_files/figure-gfm/unnamed-chunk-8-1.png" /><!-- --></p>
<!-- ![Alt text](https://github.com/foreal17/RNA-seq-workshop/blob/master/Prep_Files/Images/Gene_lengths_weight.png) -->

<h4 id="gene-length-vs-average-expression">Gene length vs average expression<a class="headerlink" href="#gene-length-vs-average-expression" title="Permanent link">&para;</a></h4>
<p>Is there an association between gene length and expression level?</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span>
<a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a><span class="nf">data.frame</span><span class="p">(</span><span class="n">logGeneLength</span> <span class="o">=</span> <span class="nf">log2</span><span class="p">(</span><span class="n">pwf</span><span class="o">$</span><span class="n">bias.data</span><span class="p">),</span> 
<a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>           <span class="n">avgExpr</span> <span class="o">=</span> <span class="n">tt</span><span class="o">$</span><span class="n">AveExpr</span><span class="p">)</span> <span class="o">%&gt;%</span> 
<a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a>  <span class="nf">ggplot</span><span class="p">(</span><span class="n">.</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">logGeneLength</span><span class="p">,</span> <span class="n">y</span><span class="o">=</span><span class="n">avgExpr</span><span class="p">))</span> <span class="o">+</span> 
<a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a>  <span class="nf">geom_point</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">0.2</span><span class="p">)</span> <span class="o">+</span> 
<a id="__codelineno-8-6" name="__codelineno-8-6" href="#__codelineno-8-6"></a>  <span class="nf">geom_smooth</span><span class="p">(</span><span class="n">method</span><span class="o">=</span><span class="s">&#39;lm&#39;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-pathway_files/figure-gfm/unnamed-chunk-9-1.png" /><!-- --></p>
<p>How about gene length and statistical evidence supporting differential
expression?</p>
<p>(Kinda hard to see, but it is apparently there…)</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="nf">data.frame</span><span class="p">(</span><span class="n">logGeneLength</span> <span class="o">=</span> <span class="nf">log2</span><span class="p">(</span><span class="n">pwf</span><span class="o">$</span><span class="n">bias.data</span><span class="p">),</span> 
<a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>           <span class="n">negLogAdjP</span> <span class="o">=</span> <span class="o">-</span><span class="nf">log10</span><span class="p">(</span><span class="n">tt</span><span class="o">$</span><span class="n">adj.P.Val</span><span class="p">))</span> <span class="o">%&gt;%</span> 
<a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a>  <span class="nf">ggplot</span><span class="p">(</span><span class="n">.</span><span class="p">,</span> <span class="nf">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">logGeneLength</span><span class="p">,</span> <span class="n">negLogAdjP</span><span class="p">))</span> <span class="o">+</span> 
<a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a>  <span class="nf">geom_point</span><span class="p">(</span><span class="n">size</span><span class="o">=</span><span class="m">0.2</span><span class="p">)</span> <span class="o">+</span> 
<a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a>  <span class="nf">geom_smooth</span><span class="p">(</span><span class="n">method</span><span class="o">=</span><span class="s">&#39;lm&#39;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-pathway_files/figure-gfm/unnamed-chunk-10-1.png" /><!-- --></p>
<!-- ![Alt text](https://github.com/foreal17/RNA-seq-workshop/blob/master/Prep_Files/Images/GeneLength_vs_AvgExpression.png) -->

<h4 id="length-correction-in-goseq">Length correction in GOSeq<a class="headerlink" href="#length-correction-in-goseq" title="Permanent link">&para;</a></h4>
<ul>
<li>Uses “Wallenius approximation” to perform correction.</li>
<li>Essentially it is performing a weighted Fisher’s Exact Test, but
    each gene in the 2x2 data does not contribute equally to the per
    cell count - it instead contributes its weight (based on its
    length).</li>
<li>This means that a gene set (e.g., GO term) containing lots of
    significant short genes will be considered more likely to be
    enriched that a gene set with a similar proportion of long genes
    that are differentially expressed.</li>
</ul>
<p>Run GOSeq with gene length correction:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="n">GO.wall</span><span class="o">=</span><span class="nf">goseq</span><span class="p">(</span><span class="n">pwf</span><span class="p">,</span> <span class="s">&quot;sacCer1&quot;</span><span class="p">,</span> <span class="s">&quot;ensGene&quot;</span><span class="p">)</span>
</code></pre></div>
<h4 id="output-wallenius-method">Output: Wallenius method<a class="headerlink" href="#output-wallenius-method" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="nf">head</span><span class="p">(</span><span class="n">GO.wall</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##        category over_represented_pvalue under_represented_pvalue numDEInCat
## 1325 GO:0005622            1.817597e-17                        1       4298
## 7912 GO:0110165            8.200109e-12                        1       4456
## 5245 GO:0042254            9.441146e-12                        1        364
## 3817 GO:0022613            1.355385e-11                        1        439
## 5446 GO:0043226            2.516156e-11                        1       3826
## 5449 GO:0043229            4.433252e-11                        1       3817
##      numInCat                                 term ontology
## 1325     5179                        intracellular       CC
## 7912     5417           cellular anatomical entity       CC
## 5245      395                  ribosome biogenesis       BP
## 3817      482 ribonucleoprotein complex biogenesis       BP
## 5446     4609                            organelle       CC
## 5449     4599              intracellular organelle       CC
</code></pre></div>
<h4 id="p-value-adjustment">P-value adjustment<a class="headerlink" href="#p-value-adjustment" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="n">GO.wall.padj</span> <span class="o">&lt;-</span> <span class="nf">p.adjust</span><span class="p">(</span><span class="n">GO.wall</span><span class="o">$</span><span class="n">over_represented_pvalue</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s">&quot;fdr&quot;</span><span class="p">)</span>
<a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a><span class="nf">sum</span><span class="p">(</span><span class="n">GO.wall.padj</span> <span class="o">&lt;</span> <span class="m">0.05</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 45
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="n">GO.wall.sig</span> <span class="o">&lt;-</span> <span class="n">GO.wall</span><span class="o">$</span><span class="n">category</span><span class="p">[</span><span class="n">GO.wall.padj</span> <span class="o">&lt;</span> <span class="m">0.05</span><span class="p">]</span>
<a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a><span class="nf">length</span><span class="p">(</span><span class="n">GO.wall.sig</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 45
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a><span class="nf">head</span><span class="p">(</span><span class="n">GO.wall.sig</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] &quot;GO:0005622&quot; &quot;GO:0110165&quot; &quot;GO:0042254&quot; &quot;GO:0022613&quot; &quot;GO:0043226&quot;
## [6] &quot;GO:0043229&quot;
</code></pre></div>
<h4 id="filtering-by-gene-set-length">Filtering by gene set length<a class="headerlink" href="#filtering-by-gene-set-length" title="Permanent link">&para;</a></h4>
<p>We are usually not interested in pathways or ontology terms that involve
large numbers of genes (they are often rather broad terms or
mechanisms), so we can exclude these from the results (here we only
include significant GO terms that involve less than 500 genes):</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="n">GO.wall</span><span class="p">[</span><span class="n">GO.wall.padj</span> <span class="o">&lt;</span> <span class="m">0.05</span><span class="p">,</span> <span class="p">]</span> <span class="o">%&gt;%</span> <span class="nf">filter</span><span class="p">(</span><span class="n">numInCat</span> <span class="o">&lt;</span> <span class="m">500</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##      category over_represented_pvalue under_represented_pvalue numDEInCat
## 1  GO:0042254            9.441146e-12                1.0000000        364
## 2  GO:0022613            1.355385e-11                1.0000000        439
## 3  GO:0030684            3.348423e-08                1.0000000        163
## 4  GO:0016072            3.873737e-08                1.0000000        274
## 5  GO:0006364            4.532117e-08                1.0000000        259
## 6  GO:0034470            2.776978e-07                1.0000000        360
## 7  GO:0030490            9.891093e-07                0.9999998        111
## 8  GO:0030687            1.502461e-06                1.0000000         62
## 9  GO:0042274            2.616777e-06                0.9999993        134
## 10 GO:0005730            3.753501e-06                1.0000000        264
## 11 GO:0003735            3.799990e-06                1.0000000        194
## 12 GO:0022626            4.824970e-06                0.9999984        140
## 13 GO:0044391            6.732064e-06                1.0000000        202
## 14 GO:0000462            1.221128e-05                0.9999977         99
## 15 GO:0005840            1.780243e-05                0.9999942        221
## 16 GO:0002181            2.156069e-05                0.9999913        169
## 17 GO:0034660            7.782098e-05                0.9999571        409
## 18 GO:0042273            8.604282e-05                0.9999745        117
##    numInCat
## 1       395
## 2       482
## 3       171
## 4       299
## 5       282
## 6       401
## 7       116
## 8        62
## 9       143
## 10      292
## 11      221
## 12      156
## 13      231
## 14      104
## 15      254
## 16      191
## 17      467
## 18      126
##                                                                                        term
## 1                                                                       ribosome biogenesis
## 2                                                      ribonucleoprotein complex biogenesis
## 3                                                                               preribosome
## 4                                                                    rRNA metabolic process
## 5                                                                           rRNA processing
## 6                                                                          ncRNA processing
## 7                                                                    maturation of SSU-rRNA
## 8                                                      preribosome, large subunit precursor
## 9                                                        ribosomal small subunit biogenesis
## 10                                                                                nucleolus
## 11                                                       structural constituent of ribosome
## 12                                                                       cytosolic ribosome
## 13                                                                        ribosomal subunit
## 14 maturation of SSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)
## 15                                                                                 ribosome
## 16                                                                  cytoplasmic translation
## 17                                                                  ncRNA metabolic process
## 18                                                       ribosomal large subunit biogenesis
##    ontology
## 1        BP
## 2        BP
## 3        CC
## 4        BP
## 5        BP
## 6        BP
## 7        BP
## 8        CC
## 9        BP
## 10       CC
## 11       MF
## 12       CC
## 13       CC
## 14       BP
## 15       CC
## 16       BP
## 17       BP
## 18       BP
</code></pre></div>
<hr />
<h2 id="go-terms">GO terms<a class="headerlink" href="#go-terms" title="Permanent link">&para;</a></h2>
<ul>
<li>Can use the <code>GO.db</code> package to get more information about the
    significant gene sets.</li>
</ul>
<div class="highlight"><pre><span></span><code><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">GO.db</span><span class="p">)</span>
<a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a>
<a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a><span class="n">GOTERM</span><span class="p">[[</span><span class="n">GO.wall.sig</span><span class="p">[</span><span class="m">1</span><span class="p">]]]</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## GOID: GO:0005622
## Term: intracellular
## Ontology: CC
## Definition: The living contents of a cell; the matter contained within
##     (but not including) the plasma membrane, usually taken to exclude
##     large vacuoles and masses of secretory or ingested material. In
##     eukaryotes it includes the nucleus and cytoplasm.
## Synonym: internal to cell
## Synonym: protoplasm
## Synonym: nucleocytoplasm
## Synonym: protoplast
</code></pre></div>
<hr />
<p><strong>NB - the code below is demonstrating the difference between running
with and without the gene length correction that GOseq implements. You
wouldn’t usually work through this as part of a standard pathway
analysis.</strong></p>
<hr />
<h4 id="run-goseq-without-gene-length-correction">Run GOSeq without gene length correction<a class="headerlink" href="#run-goseq-without-gene-length-correction" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a><span class="n">GO.nobias</span><span class="o">=</span><span class="nf">goseq</span><span class="p">(</span><span class="n">pwf</span><span class="p">,</span> <span class="s">&quot;sacCer1&quot;</span><span class="p">,</span> <span class="s">&quot;ensGene&quot;</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s">&quot;Hypergeometric&quot;</span><span class="p">)</span>
</code></pre></div>
<p><em>Output: Hypergeomtric (Fisher) method</em></p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a><span class="nf">head</span><span class="p">(</span><span class="n">GO.nobias</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>##        category over_represented_pvalue under_represented_pvalue numDEInCat
## 1325 GO:0005622            8.650971e-29                        1       4298
## 7912 GO:0110165            3.271100e-21                        1       4456
## 5446 GO:0043226            9.826546e-16                        1       3826
## 2650 GO:0009987            1.330459e-15                        1       4146
## 5449 GO:0043229            2.060920e-15                        1       3817
## 5245 GO:0042254            7.222008e-11                        1        364
##      numInCat                       term ontology
## 1325     5179              intracellular       CC
## 7912     5417 cellular anatomical entity       CC
## 5446     4609                  organelle       CC
## 2650     5023           cellular process       BP
## 5449     4599    intracellular organelle       CC
## 5245      395        ribosome biogenesis       BP
</code></pre></div>
<h4 id="p-value-adjustment_1">P-value adjustment<a class="headerlink" href="#p-value-adjustment_1" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a><span class="n">GO.nobias.padj</span> <span class="o">&lt;-</span> <span class="nf">p.adjust</span><span class="p">(</span><span class="n">GO.nobias</span><span class="o">$</span><span class="n">over_represented_pvalue</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s">&quot;fdr&quot;</span><span class="p">)</span>
<a id="__codelineno-19-2" name="__codelineno-19-2" href="#__codelineno-19-2"></a><span class="nf">sum</span><span class="p">(</span><span class="n">GO.nobias.padj</span> <span class="o">&lt;</span> <span class="m">0.05</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 44
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a><span class="n">GO.nobias.sig</span> <span class="o">&lt;-</span> <span class="n">GO.nobias</span><span class="o">$</span><span class="n">category</span><span class="p">[</span><span class="n">GO.nobias.padj</span> <span class="o">&lt;</span> <span class="m">0.05</span><span class="p">]</span>
<a id="__codelineno-20-2" name="__codelineno-20-2" href="#__codelineno-20-2"></a><span class="nf">length</span><span class="p">(</span><span class="n">GO.nobias.sig</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 44
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a><span class="nf">head</span><span class="p">(</span><span class="n">GO.nobias.sig</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] &quot;GO:0005622&quot; &quot;GO:0110165&quot; &quot;GO:0043226&quot; &quot;GO:0009987&quot; &quot;GO:0043229&quot;
## [6] &quot;GO:0042254&quot;
</code></pre></div>
<h3 id="compare-with-and-without-adjustment">Compare with and without adjustment<a class="headerlink" href="#compare-with-and-without-adjustment" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-22-1" name="__codelineno-22-1" href="#__codelineno-22-1"></a><span class="nf">library</span><span class="p">(</span><span class="n">gplots</span><span class="p">)</span>
<a id="__codelineno-22-2" name="__codelineno-22-2" href="#__codelineno-22-2"></a><span class="nf">venn</span><span class="p">(</span><span class="nf">list</span><span class="p">(</span><span class="n">GO.wall</span><span class="o">=</span><span class="n">GO.wall.sig</span><span class="p">,</span> <span class="n">GO.nobias</span><span class="o">=</span><span class="n">GO.nobias.sig</span><span class="p">))</span>
</code></pre></div>
<p><center><img alt="image" src="../rnaseq-pathway_files/figure-gfm/GO_venn.png" /></center></p>
<p>Extract out the different parts of the Venn diagram (yes, there are
definitely better ways to do this).</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-23-1" name="__codelineno-23-1" href="#__codelineno-23-1"></a><span class="c1">## Only significant in Hypergeomtric analysis</span>
<a id="__codelineno-23-2" name="__codelineno-23-2" href="#__codelineno-23-2"></a><span class="n">onlySig.nobias</span> <span class="o">&lt;-</span> <span class="nf">setdiff</span><span class="p">(</span><span class="n">GO.nobias.sig</span><span class="p">,</span> <span class="n">GO.wall.sig</span><span class="p">)</span>
<a id="__codelineno-23-3" name="__codelineno-23-3" href="#__codelineno-23-3"></a>
<a id="__codelineno-23-4" name="__codelineno-23-4" href="#__codelineno-23-4"></a><span class="c1">## Only significant in Wallenius analysis</span>
<a id="__codelineno-23-5" name="__codelineno-23-5" href="#__codelineno-23-5"></a><span class="n">onlySig.wall</span> <span class="o">&lt;-</span> <span class="nf">setdiff</span><span class="p">(</span><span class="n">GO.wall.sig</span><span class="p">,</span> <span class="n">GO.nobias.sig</span><span class="p">)</span>
<a id="__codelineno-23-6" name="__codelineno-23-6" href="#__codelineno-23-6"></a>
<a id="__codelineno-23-7" name="__codelineno-23-7" href="#__codelineno-23-7"></a><span class="c1">## Significant in both</span>
<a id="__codelineno-23-8" name="__codelineno-23-8" href="#__codelineno-23-8"></a><span class="n">sig.wall.nobias</span> <span class="o">&lt;-</span> <span class="nf">intersect</span><span class="p">(</span><span class="n">GO.wall.sig</span><span class="p">,</span> <span class="n">GO.nobias.sig</span><span class="p">)</span>
</code></pre></div>
<h3 id="gene-lengths-and-go-term-membership">Gene lengths and GO term membership<a class="headerlink" href="#gene-lengths-and-go-term-membership" title="Permanent link">&para;</a></h3>
<p>Can also extract gene length and GO membership information.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-24-1" name="__codelineno-24-1" href="#__codelineno-24-1"></a><span class="n">len</span><span class="o">=</span><span class="nf">getlength</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">genes</span><span class="p">),</span><span class="s">&quot;sacCer1&quot;</span><span class="p">,</span><span class="s">&quot;ensGene&quot;</span><span class="p">)</span>
<a id="__codelineno-24-2" name="__codelineno-24-2" href="#__codelineno-24-2"></a>
<a id="__codelineno-24-3" name="__codelineno-24-3" href="#__codelineno-24-3"></a><span class="nf">head</span><span class="p">(</span><span class="n">len</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] 1504 1621 1543 1711 1399 1504
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-25-1" name="__codelineno-25-1" href="#__codelineno-25-1"></a><span class="n">go</span> <span class="o">=</span> <span class="nf">getgo</span><span class="p">(</span><span class="nf">names</span><span class="p">(</span><span class="n">genes</span><span class="p">),</span><span class="s">&quot;sacCer1&quot;</span><span class="p">,</span><span class="s">&quot;ensGene&quot;</span><span class="p">)</span>
<a id="__codelineno-25-2" name="__codelineno-25-2" href="#__codelineno-25-2"></a>
<a id="__codelineno-25-3" name="__codelineno-25-3" href="#__codelineno-25-3"></a><span class="nf">names</span><span class="p">(</span><span class="n">go</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">head</span><span class="p">()</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] &quot;YAL038W&quot; &quot;YOR161C&quot; &quot;YML128C&quot; &quot;YMR105C&quot; &quot;YHL021C&quot; &quot;YDR516C&quot;
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-26-1" name="__codelineno-26-1" href="#__codelineno-26-1"></a><span class="nf">class</span><span class="p">(</span><span class="n">go</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] &quot;list&quot;
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-27-1" name="__codelineno-27-1" href="#__codelineno-27-1"></a><span class="nf">head</span><span class="p">(</span><span class="n">go</span><span class="p">[[</span><span class="m">1</span><span class="p">]])</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] &quot;GO:0006082&quot; &quot;GO:0008150&quot; &quot;GO:0008152&quot; &quot;GO:0009987&quot; &quot;GO:0019752&quot;
## [6] &quot;GO:0032787&quot;
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-28-1" name="__codelineno-28-1" href="#__codelineno-28-1"></a><span class="nf">head</span><span class="p">(</span><span class="n">go</span><span class="p">[[</span><span class="m">2</span><span class="p">]])</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code>## [1] &quot;GO:0006810&quot; &quot;GO:0008150&quot; &quot;GO:0009987&quot; &quot;GO:0051179&quot; &quot;GO:0051234&quot;
## [6] &quot;GO:0055085&quot;
</code></pre></div>
<hr />
<h4 id="getting-fancy">Getting fancy…<a class="headerlink" href="#getting-fancy" title="Permanent link">&para;</a></h4>
<p>Figure out which genes are in the significant GO groups, and then gets
their lengths.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-29-1" name="__codelineno-29-1" href="#__codelineno-29-1"></a><span class="n">lengths.onlySig.nobias</span> <span class="o">&lt;-</span> <span class="nf">list</span><span class="p">()</span>
<a id="__codelineno-29-2" name="__codelineno-29-2" href="#__codelineno-29-2"></a>
<a id="__codelineno-29-3" name="__codelineno-29-3" href="#__codelineno-29-3"></a><span class="nf">for</span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">onlySig.nobias</span><span class="p">)){</span>
<a id="__codelineno-29-4" name="__codelineno-29-4" href="#__codelineno-29-4"></a>  <span class="n">inGo</span> <span class="o">&lt;-</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">go</span><span class="p">,</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>  <span class="n">onlySig.nobias</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">%in%</span> <span class="n">x</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">unlist</span><span class="p">()</span>
<a id="__codelineno-29-5" name="__codelineno-29-5" href="#__codelineno-29-5"></a>  <span class="n">lengths.onlySig.nobias</span><span class="p">[[</span><span class="n">i</span><span class="p">]]</span> <span class="o">&lt;-</span> <span class="n">len</span><span class="p">[</span><span class="n">inGo</span><span class="p">]</span>
<a id="__codelineno-29-6" name="__codelineno-29-6" href="#__codelineno-29-6"></a><span class="p">}</span>
<a id="__codelineno-29-7" name="__codelineno-29-7" href="#__codelineno-29-7"></a>
<a id="__codelineno-29-8" name="__codelineno-29-8" href="#__codelineno-29-8"></a><span class="n">lengths.onlySig.wall</span> <span class="o">&lt;-</span> <span class="nf">list</span><span class="p">()</span>
<a id="__codelineno-29-9" name="__codelineno-29-9" href="#__codelineno-29-9"></a>
<a id="__codelineno-29-10" name="__codelineno-29-10" href="#__codelineno-29-10"></a><span class="nf">for</span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">onlySig.wall</span><span class="p">)){</span>
<a id="__codelineno-29-11" name="__codelineno-29-11" href="#__codelineno-29-11"></a>  <span class="n">inGo</span> <span class="o">&lt;-</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">go</span><span class="p">,</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>  <span class="n">onlySig.wall</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">%in%</span> <span class="n">x</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">unlist</span><span class="p">()</span>
<a id="__codelineno-29-12" name="__codelineno-29-12" href="#__codelineno-29-12"></a>  <span class="n">lengths.onlySig.wall</span><span class="p">[[</span><span class="n">i</span><span class="p">]]</span> <span class="o">&lt;-</span> <span class="n">len</span><span class="p">[</span><span class="n">inGo</span><span class="p">]</span>
<a id="__codelineno-29-13" name="__codelineno-29-13" href="#__codelineno-29-13"></a><span class="p">}</span>
</code></pre></div>
<h4 id="significant-hypergeometric-vs-wallenius">Significant: Hypergeometric vs Wallenius<a class="headerlink" href="#significant-hypergeometric-vs-wallenius" title="Permanent link">&para;</a></h4>
<ul>
<li>Only Hypergeometric (pink) vs only Wallenius (blue)</li>
<li>Hypergeometric method is findings GO terms containing longer genes.</li>
</ul>
<p><div class="highlight"><pre><span></span><code><a id="__codelineno-30-1" name="__codelineno-30-1" href="#__codelineno-30-1"></a><span class="n">cols</span> <span class="o">&lt;-</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;lightpink&quot;</span><span class="p">,</span> <span class="s">&quot;lightblue&quot;</span><span class="p">),</span> <span class="nf">c</span><span class="p">(</span><span class="m">10</span><span class="p">,</span><span class="m">7</span><span class="p">))</span>
<a id="__codelineno-30-2" name="__codelineno-30-2" href="#__codelineno-30-2"></a><span class="nf">boxplot</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">lengths.onlySig.nobias</span><span class="p">,</span> <span class="n">lengths.onlySig.wall</span><span class="p">),</span> <span class="n">col</span><span class="o">=</span><span class="n">cols</span><span class="p">)</span>
</code></pre></div>
<img alt="image" src="../rnaseq-pathway_files/figure-gfm/GO_boxplot1.png" /></p>
<h4 id="all-significant-go-terms">All significant GO terms<a class="headerlink" href="#all-significant-go-terms" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-31-1" name="__codelineno-31-1" href="#__codelineno-31-1"></a><span class="n">lengths.sig.wall.nobias</span> <span class="o">&lt;-</span> <span class="nf">list</span><span class="p">()</span>
<a id="__codelineno-31-2" name="__codelineno-31-2" href="#__codelineno-31-2"></a>
<a id="__codelineno-31-3" name="__codelineno-31-3" href="#__codelineno-31-3"></a><span class="nf">for</span><span class="p">(</span><span class="n">i</span> <span class="n">in</span> <span class="m">1</span><span class="o">:</span><span class="nf">length</span><span class="p">(</span><span class="n">sig.wall.nobias</span><span class="p">)){</span>
<a id="__codelineno-31-4" name="__codelineno-31-4" href="#__codelineno-31-4"></a>  <span class="n">inGo</span> <span class="o">&lt;-</span> <span class="nf">lapply</span><span class="p">(</span><span class="n">go</span><span class="p">,</span> <span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span>  <span class="n">sig.wall.nobias</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">%in%</span> <span class="n">x</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">unlist</span><span class="p">()</span>
<a id="__codelineno-31-5" name="__codelineno-31-5" href="#__codelineno-31-5"></a>  <span class="n">lengths.sig.wall.nobias</span><span class="p">[[</span><span class="n">i</span><span class="p">]]</span> <span class="o">&lt;-</span> <span class="n">len</span><span class="p">[</span><span class="n">inGo</span><span class="p">]</span>
<a id="__codelineno-31-6" name="__codelineno-31-6" href="#__codelineno-31-6"></a><span class="p">}</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-32-1" name="__codelineno-32-1" href="#__codelineno-32-1"></a><span class="n">cols</span> <span class="o">&lt;-</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;lightpink&quot;</span><span class="p">,</span> <span class="nf">grey</span><span class="p">(</span><span class="m">0.7</span><span class="p">),</span> <span class="s">&quot;lightblue&quot;</span><span class="p">),</span> <span class="nf">c</span><span class="p">(</span><span class="m">10</span><span class="p">,</span><span class="m">37</span><span class="p">,</span><span class="m">7</span><span class="p">))</span>
<a id="__codelineno-32-2" name="__codelineno-32-2" href="#__codelineno-32-2"></a>
<a id="__codelineno-32-3" name="__codelineno-32-3" href="#__codelineno-32-3"></a><span class="n">avgLength</span> <span class="o">&lt;-</span> <span class="nf">lapply</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">lengths.onlySig.nobias</span><span class="p">,</span> <span class="n">lengths.sig.wall.nobias</span><span class="p">,</span> <span class="n">lengths.onlySig.wall</span><span class="p">),</span>
<a id="__codelineno-32-4" name="__codelineno-32-4" href="#__codelineno-32-4"></a>                    <span class="n">median</span><span class="p">)</span> <span class="o">%&gt;%</span> <span class="nf">unlist</span><span class="p">()</span>
<a id="__codelineno-32-5" name="__codelineno-32-5" href="#__codelineno-32-5"></a>
<a id="__codelineno-32-6" name="__codelineno-32-6" href="#__codelineno-32-6"></a><span class="n">oo</span> <span class="o">&lt;-</span> <span class="nf">order</span><span class="p">(</span><span class="n">avgLength</span><span class="p">,</span> <span class="n">decreasing</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-33-1" name="__codelineno-33-1" href="#__codelineno-33-1"></a><span class="nf">boxplot</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">lengths.onlySig.nobias</span><span class="p">,</span> <span class="n">lengths.sig.wall.nobias</span><span class="p">,</span> <span class="n">lengths.onlySig.wall</span><span class="p">)[</span><span class="n">oo</span><span class="p">],</span>
<a id="__codelineno-33-2" name="__codelineno-33-2" href="#__codelineno-33-2"></a>        <span class="n">col</span><span class="o">=</span><span class="n">cols</span><span class="p">[</span><span class="n">oo</span><span class="p">],</span> <span class="n">ylab</span><span class="o">=</span><span class="s">&quot;Gene Length&quot;</span><span class="p">,</span> <span class="n">xlab</span> <span class="o">=</span> <span class="s">&quot;GO term&quot;</span><span class="p">)</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-pathway_files/figure-gfm/unnamed-chunk-26-1.png" /><!-- --></p>
<!-- ![Alt text](https://github.com/foreal17/RNA-seq-workshop/blob/master/Prep_Files/Images/GO_All_significant.png) -->

<h4 id="gene-length-versus-p-value">Gene length versus P-value<a class="headerlink" href="#gene-length-versus-p-value" title="Permanent link">&para;</a></h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-34-1" name="__codelineno-34-1" href="#__codelineno-34-1"></a><span class="n">avgLength.wall</span> <span class="o">&lt;-</span> <span class="nf">lapply</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">lengths.onlySig.wall</span><span class="p">,</span> <span class="n">lengths.sig.wall.nobias</span><span class="p">),</span> <span class="n">median</span><span class="p">)</span>
<a id="__codelineno-34-2" name="__codelineno-34-2" href="#__codelineno-34-2"></a>
<a id="__codelineno-34-3" name="__codelineno-34-3" href="#__codelineno-34-3"></a><span class="n">avgLength.nobias</span> <span class="o">&lt;-</span> <span class="nf">lapply</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">lengths.onlySig.nobias</span><span class="p">,</span> <span class="n">lengths.sig.wall.nobias</span><span class="p">),</span> <span class="n">median</span><span class="p">)</span>
<a id="__codelineno-34-4" name="__codelineno-34-4" href="#__codelineno-34-4"></a>
<a id="__codelineno-34-5" name="__codelineno-34-5" href="#__codelineno-34-5"></a><span class="n">cols</span> <span class="o">&lt;-</span> <span class="nf">rep</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;blue&quot;</span><span class="p">,</span> <span class="s">&quot;lightblue&quot;</span><span class="p">,</span> <span class="s">&quot;red&quot;</span><span class="p">,</span><span class="s">&quot;lightpink&quot;</span><span class="p">),</span>
<a id="__codelineno-34-6" name="__codelineno-34-6" href="#__codelineno-34-6"></a>            <span class="nf">c</span><span class="p">(</span><span class="nf">length</span><span class="p">(</span><span class="n">lengths.onlySig.wall</span><span class="p">),</span> <span class="nf">length</span><span class="p">(</span><span class="n">lengths.sig.wall.nobias</span><span class="p">),</span>
<a id="__codelineno-34-7" name="__codelineno-34-7" href="#__codelineno-34-7"></a>              <span class="nf">length</span><span class="p">(</span><span class="n">lengths.onlySig.nobias</span><span class="p">),</span> <span class="nf">length</span><span class="p">(</span><span class="n">lengths.sig.wall.nobias</span><span class="p">)))</span>
<a id="__codelineno-34-8" name="__codelineno-34-8" href="#__codelineno-34-8"></a>
<a id="__codelineno-34-9" name="__codelineno-34-9" href="#__codelineno-34-9"></a><span class="nf">plot</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">avgLength.wall</span><span class="p">,</span> <span class="n">avgLength.nobias</span><span class="p">),</span> 
<a id="__codelineno-34-10" name="__codelineno-34-10" href="#__codelineno-34-10"></a>     <span class="o">-</span><span class="nf">log</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="n">GO.nobias.padj</span><span class="p">[</span><span class="n">GO.nobias.padj</span> <span class="o">&lt;</span> <span class="m">0.05</span><span class="p">],</span> <span class="n">GO.wall.padj</span><span class="p">[</span><span class="n">GO.wall.padj</span> <span class="o">&lt;</span> <span class="m">0.05</span><span class="p">])),</span> 
<a id="__codelineno-34-11" name="__codelineno-34-11" href="#__codelineno-34-11"></a>     <span class="n">col</span><span class="o">=</span><span class="n">cols</span><span class="p">,</span> <span class="n">pch</span><span class="o">=</span><span class="m">16</span><span class="p">,</span> <span class="n">xlab</span><span class="o">=</span><span class="s">&quot;Median Gene Length&quot;</span><span class="p">,</span> <span class="n">ylab</span> <span class="o">=</span><span class="s">&quot;-log(FDR adj-pval)&quot;</span><span class="p">)</span>
<a id="__codelineno-34-12" name="__codelineno-34-12" href="#__codelineno-34-12"></a><span class="nf">legend</span><span class="p">(</span><span class="s">&#39;topright&#39;</span><span class="p">,</span> <span class="nf">c</span><span class="p">(</span><span class="s">&quot;Only sig in NoBias&quot;</span><span class="p">,</span> <span class="s">&quot;Sig in both (nobias adjp)&quot;</span><span class="p">,</span> 
<a id="__codelineno-34-13" name="__codelineno-34-13" href="#__codelineno-34-13"></a>                     <span class="s">&quot;Sig in both (wal adjp)&quot;</span><span class="p">,</span> <span class="s">&quot;Only sig in Wall&quot;</span><span class="p">),</span> 
<a id="__codelineno-34-14" name="__codelineno-34-14" href="#__codelineno-34-14"></a>       <span class="n">fill</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;red&quot;</span><span class="p">,</span> <span class="s">&quot;pink&quot;</span><span class="p">,</span> <span class="s">&quot;lightblue&quot;</span><span class="p">,</span> <span class="s">&quot;blue&quot;</span><span class="p">))</span>
</code></pre></div>
<p><img alt="image" src="../rnaseq-pathway_files/figure-gfm/GeneLength_vs_p-value.png" /><!-- --></p>
<hr />
<h2 id="summary">Summary<a class="headerlink" href="#summary" title="Permanent link">&para;</a></h2>
<ul>
<li>Once we’ve generated count data, there are a number of ways to
    perform a differential expression analysis.</li>
<li>DESeq2 and edgeR model the count data, and assume a Negative
    Binomial distribution</li>
<li>Limma transforms (and logs) the data and assumes normality</li>
<li>Here we’ve seen that these three approaches give quite similar
    results.</li>
<li>For Gene Set analysis, gene length needs to be accounted for, since
    longer transcripts are more likely to be found to be differentially
    expressed.<ul>
<li>GOSeq adjusts for transcript length to take this into account.</li>
<li>It is also possible to use GOSeq with other types of annotation
    (e.g., Reactome or KEGG pathways).</li>
</ul>
</li>
</ul>
<hr />
<p align="center"><b><a class="btn" href="https://genomicsaotearoa.github.io/RNA-seq-workshop/" style="background: var(--bs-dark);font-weight:bold">Back to homepage</a></b></p>

              
            </article>
            
          </div>
        </div>
        
          <a href="#" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </a>
        
      </main>
      
        <footer class="md-footer">
  
    
    <nav class="md-footer__inner md-grid" aria-label="Footer" >
      
        
        <a href="../5_rnaseq-diffexp/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Differential Expression Analysis" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Differential Expression Analysis
            </div>
          </div>
        </a>
      
      
        
        <a href="../supp_1/" class="md-footer__link md-footer__link--next" aria-label="Next: Supplementary 1 - A Guide to RNA-Seq" rel="next">
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Next
              </span>
              Supplementary 1 - A Guide to RNA-Seq
            </div>
          </div>
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
          </div>
        </a>
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top"], "search": "../assets/javascripts/workers/search.b97dbffb.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.6c7ad80a.min.js"></script>
      
        <script src="../javascripts/mathjax.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>